# Visium Part I

Joselyn Cristina Ch√°vez Fuentes

August 5th 2024

## Introduction to spatial dataset

10X genomics recently launched a new platform to obtain spatial expression data using a Visium Spatial Gene Expression slide.

The Visium brain data to run this tutorial can be found [here](https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Adult_Mouse_Brain)

## Create the Giotto object

```{r, eval=FALSE}
library(Giotto)

## Set instructions
results_folder <- "results/"

my_python_path <- NULL

instrs <- createGiottoInstructions(
    save_dir = results_folder,
    save_plot = TRUE,
    show_plot = FALSE,
    python_path = my_python_path
)

## Provide path to visium folder
data_path <- "data/"

## Create object directly from visium folder
visium_brain <- createGiottoVisiumObject(
    visium_dir = data_path,
    expr_data = "raw",
    png_name = "tissue_lowres_image.png",
    gene_column_index = 2,
    instructions = instrs
)
```


## Subset on spots that were covered by tissue

```{r, eval=FALSE}
## show plot
spatPlot2D(
    gobject = visium_brain,
    cell_color = "in_tissue",
    point_size = 2,
    cell_color_code = c("0" = "lightgrey", "1" = "blue"),
    show_image = TRUE,
    image_name = "image"
)

metadata <- pDataDT(visium_brain)
in_tissue_barcodes <- metadata[in_tissue == 1]$cell_ID
visium_brain <- subsetGiotto(visium_brain,
    cell_ids = in_tissue_barcodes
)
```


## Quality control

- Statistics

```{r, eval=FALSE}
visium_brain_statistics <- addStatistics(gobject = visium_brain)

## visualize
spatPlot2D(gobject = visium_brain_statistics, 
           show_image = TRUE, 
           point_alpha = 0.7,
           cell_color = 'nr_feats', 
           color_as_factor = FALSE)
```


## Filtering

```{r, eval=FALSE}
visium_brain <- filterGiotto(
    gobject = visium_brain,
    expression_threshold = 1,
    feat_det_in_min_cells = 50,
    min_det_feats_per_cell = 1000,
    expression_values = c("raw"),
    verbose = TRUE
)
```


## Normalization


```{r, eval=FALSE}
visium_brain <- normalizeGiotto(
    gobject = visium_brain,
    scalefactor = 6000,
    verbose = TRUE
)
```

```{r, eval=FALSE}
visium_brain <- addStatistics(gobject = visium_brain)

## visualize
spatPlot2D(gobject = visium_brain, 
           show_image = TRUE, 
           point_alpha = 0.7,
           cell_color = 'nr_feats', 
           color_as_factor = FALSE)
```

## Feature selection
 
 
### Highly Variable Features:

- loess regression

```{r, eval=FALSE}
visium_brain <- calculateHVF(gobject = visium_brain, 
                             method = "cov_loess",
                             save_plot = TRUE,
                             return_plot = TRUE,
                             default_save_name = "HVFplot_loess")
```

- binned

```{r, eval=FALSE}
visium_brain <- calculateHVF(gobject = visium_brain, 
                             method = "cov_groups",
                             save_plot = TRUE,
                             return_plot = TRUE,
                             default_save_name = "HVFplot_binned")
```

- pearson residuals

```{r, eval=FALSE}
visium_brain <- calculateHVF(gobject = visium_brain, 
                             method = "var_p_resid",
                             save_plot = TRUE,
                             return_plot = TRUE,
                             default_save_name = "HVFplot_pearson")
```

### Spatial variable genes

```{r, eval=FALSE}
my_spatial_genes <- c("gene1", "gene2", "gene3")
```


## Dimension Reduction

- PCA

Default

```{r, eval=FALSE}
visium_brain <- runPCA(gobject = visium_brain)
```

Using specific genes

```{r, eval=FALSE}
visium_brain <- runPCA(gobject = visium_brain,
                       feats_to_use = my_spatial_genes)
```

Screeplot 

```{r, eval=FALSE}
screePlot(visium_brain, ncp = 30)
```

Visualization

```{r, eval=FALSE}
plotPCA(visium_brain)
```


- UMAP

```{r, eval=FALSE}
visium_brain <- runUMAP(visium_brain, 
                        dimensions_to_use = 1:10)
```

```{r, eval=FALSE}
plotUMAP(gobject = visium_brain)
```

- t-SNE

```{r, eval=FALSE}
visium_brain <- runtSNE(visium_brain, 
                        dimensions_to_use = 1:10)
```

```{r, eval=FALSE}
plotTSNE(gobject = visium_brain)
```

## Clustering


### Non-spatial

- Create a sNN network (default)

```{r, eval=FALSE}
visium_brain <- createNearestNetwork(gobject = visium_brain, 
                                     dimensions_to_use = 1:10, 
                                     k = 15)
```

- Create a kNN network

```{r, eval=FALSE}
visium_brain <- createNearestNetwork(gobject = visium_brain, 
                                     dimensions_to_use = 1:10, 
                                     k = 15,
                                     type = "kNN")
```

- Hierarchical clustering

- Calculate Leiden clustering

```{r, eval=FALSE}
visium_brain <- doLeidenCluster(gobject = visium_brain, 
                                resolution = 0.4, 
                                n_iterations = 1000)
```

- Visualization

```{r, eval=FALSE}
plotPCA(gobject = visium_brain,
        cell_color = 'leiden_clus')
```

```{r, eval=FALSE}
plotUMAP(gobject = visium_brain,
         cell_color = 'leiden_clus', 
         show_NN_network = FALSE, 
         point_size = 2.5)
```
 
```{r, eval=FALSE}
plotUMAP(gobject = visium_brain,
         cell_color = 'leiden_clus', 
         show_NN_network = TRUE, 
         point_size = 2.5)
```
 
```{r, eval=FALSE}
plotTSNE(gobject = visium_brain,
         cell_color = 'leiden_clus')
```

Dimension plots grouped by cluster

```{r, eval=FALSE}
spatPlot2D(visium_brain, 
           cell_color = 'leiden_clus')
```

### Spatial

- Spatial variable genes
- Spatial co-expression modules
