# Xenium

Jiaji George Chen

August 6th 2024

## Introduction to spatial dataset

```{r, echo=FALSE, out.width="80%", fig.align='center'}
knitr::include_graphics("img/02_session3/Xenium-General-Workflow-1024x709.png")
```

This is the 10X Xenium FFPE Human Lung Cancer dataset. Xenium captures
individual transcript detections with a spatial resolution of 100s of
nanometers, providing an extremely highly resolved subcellular spatial
dataset. This particular dataset also showcases their newly released
multimodal cell segmentation outputs.

Previous Xenium datasets provided nuclear segmentations based on DAPI
then generated cell

The Xenium Human Multi-Tissue and Cancer Panel (377) genes was used. The
exported data is from their Xenium Onboard Analysis v2.0.0 pipeline.

The data to run this tutorial can be found
[here](https://www.10xgenomics.com/datasets/preview-data-ffpe-human-lung-cancer-with-xenium-multimodal-cell-segmentation-1-standard)

The files can be downloaded by running the following in the bash
terminal:

```{bash, eval=FALSE}
# Input Files

wget https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_humanLung_Cancer_FFPE/Xenium_V1_humanLung_Cancer_FFPE_gene_panel.json
wget https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_humanLung_Cancer_FFPE/Xenium_V1_humanLung_Cancer_FFPE_he_image.ome.tif
wget https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_humanLung_Cancer_FFPE/Xenium_V1_humanLung_Cancer_FFPE_he_imagealignment.csv

# Output Files
wget https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_humanLung_Cancer_FFPE/Xenium_V1_humanLung_Cancer_FFPE_outs.zip
wget https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_humanLung_Cancer_FFPE/Xenium_V1_humanLung_Cancer_FFPE_xe_outs.zip

```

-   *Input Files* are additional files that are not part of the output
    bundles, usually some raw files the pipeline uses or extra info.
-   *Output Files* are the directories that are exported by the Xenium
    pipeline that can be expected from every run.

## Additional install requirements

-   arrow is needed
-   requires ZTSD codec to open these parquets.



```{r, eval=FALSE}
save_dir <- "~/Documents/GitHub/giotto_workshop_2024/img/02_session3"
```

## Read in raw data

- Transcript coordinates
- Polygon coordinates
- Visualizations

- affine mapping?

## Overlap txs & polygons

- Typical aggregated workflow
- Feature/molecule specific analysis
- Visualizations


## Transcript enrichment GSEA


text

## Spatial location analysis

- Spatial cell type co-localization analysis
- Spatial niche analysis
- Spatial niche trajectory analysis

## Visualizations















zarr_exp <- "https://scc-ondemand1.bu.edu/pun/sys/dashboard/files/fs//projectnb/rd-spat/DATA/Public_data/Spatial/Multiplexing_RNA/xenium/v2.0_pipeline/FFPE_LungCancer/cell_features"

datadir <- "/projectnb2/rd-spat/DATA/Public_data/Spatial/Multiplexing_RNA/xenium/v2.0_pipeline/FFPE_LungCancer/"

manifest <- list.files(
  datadir, full.names = TRUE,
) |> as.list()
names(manifest) <- list.files(datadir)
force(names(manifest))
  
tx_path <- manifest$transcripts.csv.gz
poly_cell_path <- manifest$cell_boundaries.csv.gz
poly_nuc_path <- manifest$nucleus_boundaries.csv.gz
    
image_path <- manifest$Xenium_V1_humanLung_Cancer_FFPE_he_image.ome.tif
aff_path <- manifest$Xenium_V1_humanLung_Cancer_FFPE_he_imagealignment.csv

cell <- manifest$cells.csv.gz

## ----- exploration

## IMAGE 

aff <- data.table::fread(aff_path)

reticulate::source_python(file = "/projectnb/rd-spat/HOME/george/scripts/tifffile_convert.py")
GiottoUtils::package_check("os", repository = "pip:os")
GiottoUtils::package_check("multiprocessing", repository = "pip:multiprocessing")
GiottoUtils::package_check("tifffile", repository = "pip:tifffile")

out_tif <- "/projectnb/rd-spat/HOME/george/tmp/tifffile_testing/test.tif"

## ometif_2_tif(image_path, output_file = out_tif, chunk_rows = chunk_rows, overwrite = TRUE)

ometif_2_tif(image_path, output_file = out_tif, overwrite = TRUE)
test <- createGiottoLargeImage(out_tif)

n_before <- .nr(tx_arrow)
tx_arrow <- dplyr::filter(tx_arrow, qv > qv_threshold)
n_after <- .nr(tx_arrow)

out_if_fmt <- "/projectnb/rd-spat/HOME/george/tmp/tifffile_testing/if%d.tif"
source_if_fmt <- "/projectnb2/rd-spat/DATA/Public_data/Spatial/Multiplexing_RNA/xenium/v2.0_pipeline/FFPE_LungCancer//morphology_focus/morphology_focus_000%d.ome.tif"

ometif_2_tif(sprintf(source_if_fmt, 0), output_file = sprintf(out_if_fmt, 0), overwrite = TRUE)
ometif_2_tif(sprintf(source_if_fmt, 1), output_file = sprintf(out_if_fmt, 1), overwrite = TRUE)
ometif_2_tif(sprintf(source_if_fmt, 2), output_file = sprintf(out_if_fmt, 2), overwrite = TRUE)
ometif_2_tif(sprintf(source_if_fmt, 3), output_file = sprintf(out_if_fmt, 3), overwrite = TRUE)

img_if_list <- lapply(
  sprintf(out_if_fmt, 0:3),
  createGiottoLargeImage
)




## Example usage

input_file = "path/to/input/file.ome.tif"
output_file = "path/to/output/file.tif"
chunk_size = (1000, 1000)  # Adjust the chunk size as needed
num_processes = 4  # Adjust the number of processes based on available CPU cores
 
ometif_2_tif(input_file, output_file, chunk_size, num_processes)


## TRANSCRIPTS 

tx <- data.table::fread(tx_path, colClasses = c(transcript_id = "character"))
poly_cell_dt <- data.table::fread(poly_cell_path)
poly_nuc_dt <- data.table::fread(poly_nuc_path)


gpoints_list <- GiottoClass::createGiottoPoints(
  tx,
  x_colname = "x_location",
  y_colname = "y_location",
  feat_ID_colname = "feature_name",
  feat_type = c("rna", "NegControlProbe", "UnassignedCodeword", "NegControlCodeword"),
  split_keyword = list("NegControlProbe", "UnassignedCodeword", "NegControlCodeword")
)

gp_filter <- gpoints_list$rna[gpoints_list$rna$qv >= 20]

##  where are the issue TX at?

plot(gp_filter@spatVector, values = gp_filter$qv, type = "continuous", cex = 0.1, col = GiottoVisuals::getColors("magma"))

library(ggplot2)
type_qv <- data.table::rbindlist(
  list(
    data.table::data.table(type = "genes", qv =  gpoints_list$rna$qv),
    data.table::data.table(type = "NegControlProbe", qv = gpoints_list$NegControlProbe$qv),
    data.table::data.table(type = "UnassignedCodeword", qv = gpoints_list$UnassignedCodeword$qv),
    data.table::data.table(type = "NegControlCodeword", qv = gpoints_list$NegControlCodeword$qv)
  )
)
ggplot(data = type_qv, aes(x = type, y = qv, fill = type)) +
  geom_violin()


## POLYS 

gpoly_cell <- createGiottoPolygon(poly_cell_dt, name = "cell")
gpoly_nuc <- createGiottoPolygon(poly_nuc_dt, name = "nucleus")

mini_ext <- ext(6500,7500, 1300, 2000)

plot(gpoly_cell, col = "magenta", ext = mini_ext, background = "black")
plot(gpoly_nuc, add = T, col = "cyan", ext = mini_ext)

plot(gp_filter, add = T, col = "yellow", alpha = 0.3, ext = mini_ext, raster = F)


## giotto 

g <- giotto()
g <- setGiotto(g, gp_filter)
g <- setGiotto(g, gpoly_cell)
g <- setGiotto(g, gpoly_nuc)


g <- calculateOverlap(
  g,
  feat_info = "rna",
  spatial_info = "cell",
  return_gobject = T
)

g <- overlapToMatrix(
  g,
  poly_info = "cell",
  feat_info = "rna",
  return_gobject = TRUE
)

savedir <- "/projectnb2/rd-spat/HOME/george/projects/xenium_processing/lung_test/"
instructions(g, "save_dir") <- savedir
instructions(g, "save_plot") <- TRUE
instructions(g, "return_plot") <- FALSE
instructions(g, "show_plot") <- FALSE


g <- addStatistics(g, expression_values = "raw")
spatPlot2D(g, cell_color = "total_expr", color_as_factor = FALSE,
           point_shape = "no_border", point_size = 0.1, gradient_style = "s",
           background_color = "black",
           save_param = list(
             base_width = 15,
             base_height = 6
           ))

hist(pDataDT(g)$nr_feats)


filterCombinations(
  g,
  feat_det_in_min_cells = c(100, 200, 200),
  min_det_feats_per_cell = c(10, 20, 30),
  expression_thresholds = 1
)

g <- filterGiotto(
  g,
  feat_det_in_min_cells = 100,
  min_det_feats_per_cell = 10,
  expression_threshold = 1
)

g <- normalizeGiotto(g)

## no HVF calc since there are not that many different gene species



g <- runPCA(gobject = g,
           spat_unit = 'cell',
           expression_values = 'scaled',
           feats_to_use = NULL,
           scale_unit = F,
           center = F)

screePlot(g, ncp = 20)


my_colors <- c("#6D0C00", "magenta", "#FFBBCE", "#FFDDDF", "#FFEEEE", "#FFFEFE", "white", "white")


plotPCA(g,
        cell_color = "total_expr",
        color_as_factor = FALSE,
        cell_color_gradient = my_colors,
        point_shape = "no_border",
        gradient_style = "s",
        background = "black",
        point_size = 0.1,
        save_param = list(
          save_name = "pca_expr"
        )
)



g <- runUMAPprojection(
  g, dimensions_to_use = 1:15,
  random_subset = 10000
)

dimPlot2D(
  g,
  dim_reduction_name = "umap.projection",
  dim_reduction_to_use = "umap",
  cell_color = "total_expr",
  cell_color_gradient = my_colors,
  color_as_factor = FALSE,
  point_shape = "no_border",
  gradient_style = "s",
  background = "black",
  point_size = 0.1,
  save_param = list(
    save_name = "umap.projection_expr"
  )
)

g <- createNearestNetwork(g)

## set seed for reproducibility

set.seed(123)
g_mini <- subsetGiotto(g, cell_ids = sample(spatIDs(g), size = 10000))

g_mini <- createNearestNetwork(g_mini)

g_mini <- doLeidenCluster(g_mini, resolution = 1.4)

 dimPlot2D(
   g_mini,
   spat_unit = "cell",
   feat_type = "rna",
   cell_color = "leiden_clus",
   dim_reduction_name = "umap.projection",
   dim_reduction_to_use = "umap",
   save_param = list(
     save_name = "mini_umap_1.2"
   )
 )

clus_color <- getColors("Vivid", n = 17)[]
clus_color[9] <- "lightgrey"
clus_color[5] <- "yellow"
clus_color[3] <- "#00AAFF"
clus_color[6] <- "darkgreen"
clus_color <- clus_color[c(2, 5, 1, 7, 6, 3, 4, 9, 8, 10, 12, 11, 14, 13, 15, 16, 17)]

         
clus_color <- c("#00AAFF", "#E58606", "#B763A7", "#437478", "#87727B",
                "darkgreen", "#56BC9D", "#5668AF", "yellow", "#B99F3A",
                "#E5625E", "#448DAE", "#A5AA99", "#D8497A", "#93538D",
                "#BF5C91", "lightgrey")



dimPlot2D(
  g_mini,
  spat_unit = "cell",
  feat_type = "rna",
  cell_color = "leiden_clus", 
  cell_color_code = clus_color,
  dim_reduction_name = "umap.projection", 
  dim_reduction_to_use = "umap",
  save_param = list(
    save_name = "mini_umap_1.4"
  )
)

## 1.4 seems like a good res

project values back via kNN classifier (FNN::knn)


g <- doClusterProjection(
  target_gobject = g,
  target_cluster_label_name = "leiden_clus",
  source_gobject = g_mini, # res 1.4
  source_cluster_labels = "leiden_clus"
)

dimPlot2D(g, cell_color = "leiden_clus",
          cell_color_code = clus_color[c(1, 10:17, 2:9)],
          dim_reduction_to_use = "umap",
          dim_reduction_name = "umap.projection",
          save_param = list(
            save_name = "umap.project_1.4"
          ))

cellmeta <- pDataDT(g)
cellmeta[, major_clus := leiden_clus]
cellmeta[major_clus %in% c(1, 7, 6, 8, 3, 5), major_clus := "combined_1"]
m_clus <- cellmeta[, c("cell_ID", "major_clus")]
g <- addCellMetadata(g, new_metadata = m_clus, by_column = T)

spatPlot2D(g, cell_color = "leiden_clus",
           point_size = 0.1,
           point_shape = "no_border",
           cell_color_code = clus_color[c(1, 10:17, 2:9)],
           background = "black",
           save_param = list(
             base_width = 15,
             base_height = 6,
             save_name = "spat_leiden"
           ))

 spatPlot2D(g, cell_color = "major_clus",
            point_size = 0.1,
            point_shape = "no_border",
            cell_color_code = clus_color[c(1, 10:17, 2:9)][c(1:3,5:9, 14, 17, 11)],
            background = "black",
            save_param = list(
              base_width = 15,
              base_height = 6,
              save_name = "spat_major"
            ))

spatFeatPlot2D(g, feats = c("MKI67", "CD3E", "CD4", "CD8A", "CD19", "CD79A"),
               point_size = 0.1,
               point_shape = "no_border",
               background = "black",
               cow_n_col = 1L,
               gradient_style = "s",
               save_param = list(
                 base_width = 15,
                 base_height = 30
               ))

spatFeatPlot2D(g, feats = c("CD68", "CD27", "SFTA2", "KRT7", "SERPINB3", "SOX2", "SMOC"),
               point_size = 0.1,
               point_shape = "no_border",
               background = "black",
               cow_n_col = 1L,
               gradient_style = "s",
               save_param = list(
                 base_width = 15,
                 base_height = 30
               ))


## niches
g = createSpatialNetwork(
  gobject = g,
  spat_unit = 'cell',
  method = 'Delaunay'
)

g = calculateSpatCellMetadataProportions(
  gobject = g,
  spat_unit = 'cell',
  feat_type = 'rna',
  spat_network = 'Delaunay_network',
  metadata_column = 'leiden_clus',
  name = 'proportion'
)


## visualize niche-level enrichment for leiden cluster 3 in spat_unit "cell"

spatPlot2D(
  gobject = g,
  spat_unit = 'cell',
  point_size = 0.1,
  spat_enr_names = 'proportion',
  color_as_factor = FALSE,
  gradient_style = 'sequential',
  cell_color = '3',
  point_shape = "no_border",
  save_param = list(
    save_name = "niche_3",
    base_width = 15,
    base_height = 6
  )
)


prop_table = getSpatialEnrichment(
  g, spat_unit = 'cell', name = 'proportion', output = 'data.table'
)

convert the data.table to a sparse Matrix with row and colnames 
here we use a utility function to perform the operation

prop_matrix = GiottoUtils::dt_to_matrix(prop_table)



 These enrichments are essentially a measure of how many cells of each
 leiden cluster exist in the local region
 
 Using kmeans, we can classify each cell by its niche leiden cluster proportions

set.seed(12345) # set seed for kmeans
prop_kmeans = kmeans(x = prop_matrix, centers = 6, iter.max = 100, nstart = 3)
prop_kmeansDT = data.table::data.table(
  cell_ID = names(prop_kmeans$cluster), 
  niche = prop_kmeans$cluster
)

## add kmeans clustering of niches to cell metadata

g = addCellMetadata(
  g, spat_unit = 'cell' , 
  new_metadata = prop_kmeansDT, 
  by_column = TRUE,
  column_cell_ID = 'cell_ID'
)

## Spatially visualize the niches

spatPlot(gobject = g, 
         show_network = TRUE,
         network_color = 'lightgray', 
         spatial_network_name = 'Delaunay_network',
         cell_color = 'niche',
         point_size = 0.1,
         point_shape = "no_border",
         save_param = list(
           save_name = "niches",
           base_width = 15,
           base_height = 6
         ))














## Gene enrichment

scran_markers <- findScranMarkers_one_vs_all(
  g,
  cluster_column = "leiden_clus",
  expression_values = "normalized"
)

topgenes_scran = scran_markers[, head(.SD, 2), by = 'cluster']$feats

## violinplot

violinPlot(g,
           feats = unique(topgenes_scran), 
           cluster_column = 'leiden_clus',
           strip_text = 10, strip_position = 'right',
           save_param = list(base_width = 5))




