<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>63 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="63 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="63 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="63 Multi-omics integration | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina ChÃ¡vez-Fuentes, Junxiang Xu ,Edward Ruiz, Jeff Sheridan, Iqra Amin, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="image-transforms.html"/>
<link rel="next" href="interoperability-with-other-frameworks.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>2</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#presentation"><i class="fa fa-check"></i><b>2.1</b> Presentation</a></li>
<li class="chapter" data-level="2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#short-summary"><i class="fa fa-check"></i><b>2.2</b> Short summary</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#why-do-we-need-spatial-omics-technologies"><i class="fa fa-check"></i><b>2.2.1</b> Why do we need spatial omics technologies?</a></li>
<li class="chapter" data-level="2.2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-is-spatial-omics"><i class="fa fa-check"></i><b>2.2.2</b> What is spatial omics?</a></li>
<li class="chapter" data-level="2.2.3" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-main-spatial-omics-technologies"><i class="fa fa-check"></i><b>2.2.3</b> What are the main spatial omics technologies?</a></li>
<li class="chapter" data-level="2.2.4" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#other-spatial-omics-atac-seq-cutrun-lipidomics-etc"><i class="fa fa-check"></i><b>2.2.4</b> Other Spatial omics: ATAC-seq, CUT&amp;RUN, lipidomics, etc</a></li>
<li class="chapter" data-level="2.2.5" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#what-are-the-different-types-of-spatial-downstream-analyses"><i class="fa fa-check"></i><b>2.2.5</b> What are the different types of spatial downstream analyses?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>3</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#slides"><i class="fa fa-check"></i><b>3.1</b> Slides</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>3.2</b> Ecosystem</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>3.3</b> Installation + python environment</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>3.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>4</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>4.1</b> Data formats</a></li>
<li class="chapter" data-level="4.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>4.2</b> Pre-processing</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>5</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#overview"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>5.2</b> From matrix + locations</a></li>
<li class="chapter" data-level="5.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>5.3</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="5.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>5.4</b> From piece-wise</a></li>
<li class="chapter" data-level="5.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>5.5</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, â¦)</a></li>
<li class="chapter" data-level="5.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>5.6</b> Spatial plots</a></li>
<li class="chapter" data-level="5.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>5.7</b> Subsetting</a></li>
<li class="chapter" data-level="5.8" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>5.8</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>6</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>6.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="6.2" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>6.2</b> Download dataset</a></li>
<li class="chapter" data-level="6.3" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>6.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="6.4" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>6.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="6.5" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>6.5</b> Quality control</a></li>
<li class="chapter" data-level="6.6" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>6.6</b> Filtering</a></li>
<li class="chapter" data-level="6.7" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>6.7</b> Normalization</a></li>
<li class="chapter" data-level="6.8" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>6.8</b> Feature selection</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>6.8.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>6.9</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>6.9.1</b> PCA</a></li>
<li class="chapter" data-level="6.9.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>6.9.2</b> UMAP</a></li>
<li class="chapter" data-level="6.9.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>6.9.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>6.10</b> Clustering</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>6.10.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="6.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>6.10.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="6.11" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>6.11</b> Save the object</a></li>
<li class="chapter" data-level="6.12" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>6.12</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>7</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>7.1</b> Load the object</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>7.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>7.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="7.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>7.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>7.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>7.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="7.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>7.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>7.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>7.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="7.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>7.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>7.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>7.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>7.7</b> Interactive tools</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>7.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>8</b> Visium HD</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-hd.html"><a href="visium-hd.html#objective"><i class="fa fa-check"></i><b>8.1</b> Objective</a></li>
<li class="chapter" data-level="8.2" data-path="visium-hd.html"><a href="visium-hd.html#background"><i class="fa fa-check"></i><b>8.2</b> Background</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-technology"><i class="fa fa-check"></i><b>8.2.1</b> Visium HD Technology</a></li>
<li class="chapter" data-level="8.2.2" data-path="visium-hd.html"><a href="visium-hd.html#colorectal-cancer-sample"><i class="fa fa-check"></i><b>8.2.2</b> Colorectal Cancer Sample</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>8.3</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="visium-hd.html"><a href="visium-hd.html#visium-hd-output-data-format"><i class="fa fa-check"></i><b>8.3.1</b> Visium HD output data format</a></li>
<li class="chapter" data-level="8.3.2" data-path="visium-hd.html"><a href="visium-hd.html#read-in-raw-data"><i class="fa fa-check"></i><b>8.3.2</b> Read in raw data</a></li>
<li class="chapter" data-level="8.3.3" data-path="visium-hd.html"><a href="visium-hd.html#giotto-visium-hd-convenience-function"><i class="fa fa-check"></i><b>8.3.3</b> Giotto Visium HD convenience function</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="visium-hd.html"><a href="visium-hd.html#tiling-and-aggregation"><i class="fa fa-check"></i><b>8.4</b> Tiling and aggregation</a></li>
<li class="chapter" data-level="8.5" data-path="visium-hd.html"><a href="visium-hd.html#scalability-and-projection-functions"><i class="fa fa-check"></i><b>8.5</b> Scalability and projection functions</a></li>
<li class="chapter" data-level="8.6" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>8.6</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="8.7" data-path="visium-hd.html"><a href="visium-hd.html#spatial-co-expression-modules-1"><i class="fa fa-check"></i><b>8.7</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="xenium.html"><a href="xenium.html"><i class="fa fa-check"></i><b>9</b> Xenium</a>
<ul>
<li class="chapter" data-level="9.1" data-path="xenium.html"><a href="xenium.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>9.1</b> Introduction to spatial dataset</a></li>
<li class="chapter" data-level="9.2" data-path="xenium.html"><a href="xenium.html#additional-package-requirements"><i class="fa fa-check"></i><b>9.2</b> Additional package requirements</a></li>
<li class="chapter" data-level="9.3" data-path="xenium.html"><a href="xenium.html#introduce-mini-object"><i class="fa fa-check"></i><b>9.3</b> Introduce mini object</a></li>
<li class="chapter" data-level="9.4" data-path="xenium.html"><a href="xenium.html#read-in-raw-data-1"><i class="fa fa-check"></i><b>9.4</b> Read in raw data</a></li>
<li class="chapter" data-level="9.5" data-path="xenium.html"><a href="xenium.html#overlap-txs-polygons"><i class="fa fa-check"></i><b>9.5</b> Overlap txs &amp; polygons</a></li>
<li class="chapter" data-level="9.6" data-path="xenium.html"><a href="xenium.html#transcript-enrichment-gsea"><i class="fa fa-check"></i><b>9.6</b> Transcript enrichment GSEA</a></li>
<li class="chapter" data-level="9.7" data-path="xenium.html"><a href="xenium.html#spatial-location-analysis"><i class="fa fa-check"></i><b>9.7</b> Spatial location analysis</a></li>
<li class="chapter" data-level="9.8" data-path="xenium.html"><a href="xenium.html#visualizations"><i class="fa fa-check"></i><b>9.8</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="exploration.html"><a href="exploration.html"><i class="fa fa-check"></i><b>10</b> ââ exploration</a></li>
<li class="chapter" data-level="11" data-path="image.html"><a href="image.html"><i class="fa fa-check"></i><b>11</b> IMAGE</a></li>
<li class="chapter" data-level="12" data-path="ometif_2_tifimage_path-output_file-out_tif-chunk_rows-chunk_rows-overwrite-true.html"><a href="ometif_2_tifimage_path-output_file-out_tif-chunk_rows-chunk_rows-overwrite-true.html"><i class="fa fa-check"></i><b>12</b> ometif_2_tif(image_path, output_file = out_tif, chunk_rows = chunk_rows, overwrite = TRUE)</a></li>
<li class="chapter" data-level="13" data-path="example-usage.html"><a href="example-usage.html"><i class="fa fa-check"></i><b>13</b> # Example usage</a></li>
<li class="chapter" data-level="14" data-path="input_file-pathtoinputfile.ome.html"><a href="input_file-pathtoinputfile.ome.html"><i class="fa fa-check"></i><b>14</b> input_file = âpath/to/input/file.ome.tifâ</a></li>
<li class="chapter" data-level="15" data-path="output_file-pathtooutputfile.html"><a href="output_file-pathtooutputfile.html"><i class="fa fa-check"></i><b>15</b> output_file = âpath/to/output/file.tifâ</a></li>
<li class="chapter" data-level="16" data-path="chunk_size-1000-1000-adjust-the-chunk-size-as-needed.html"><a href="chunk_size-1000-1000-adjust-the-chunk-size-as-needed.html"><i class="fa fa-check"></i><b>16</b> chunk_size = (1000, 1000) # Adjust the chunk size as needed</a></li>
<li class="chapter" data-level="17" data-path="num_processes-4-adjust-the-number-of-processes-based-on-available-cpu-cores.html"><a href="num_processes-4-adjust-the-number-of-processes-based-on-available-cpu-cores.html"><i class="fa fa-check"></i><b>17</b> num_processes = 4 # Adjust the number of processes based on available CPU cores</a></li>
<li class="chapter" data-level="18" data-path="section.html"><a href="section.html"><i class="fa fa-check"></i><b>18</b> </a></li>
<li class="chapter" data-level="19" data-path="ometif_2_tifinput_file-output_file-chunk_size-num_processes.html"><a href="ometif_2_tifinput_file-output_file-chunk_size-num_processes.html"><i class="fa fa-check"></i><b>19</b> ometif_2_tif(input_file, output_file, chunk_size, num_processes)</a></li>
<li class="chapter" data-level="20" data-path="transcripts.html"><a href="transcripts.html"><i class="fa fa-check"></i><b>20</b> TRANSCRIPTS</a></li>
<li class="chapter" data-level="21" data-path="where-are-the-issue-tx-at.html"><a href="where-are-the-issue-tx-at.html"><i class="fa fa-check"></i><b>21</b> where are the issue TX at?</a></li>
<li class="chapter" data-level="22" data-path="polys.html"><a href="polys.html"><i class="fa fa-check"></i><b>22</b> POLYS</a></li>
<li class="chapter" data-level="23" data-path="giotto.html"><a href="giotto.html"><i class="fa fa-check"></i><b>23</b> giotto</a></li>
<li class="chapter" data-level="24" data-path="no-hvf-calc-since-there-are-not-that-many-different-gene-species.html"><a href="no-hvf-calc-since-there-are-not-that-many-different-gene-species.html"><i class="fa fa-check"></i><b>24</b> no HVF calc since there are not that many different gene species</a></li>
<li class="chapter" data-level="25" data-path="set-seed-for-reproducibility.html"><a href="set-seed-for-reproducibility.html"><i class="fa fa-check"></i><b>25</b> set seed for reproducibility</a></li>
<li class="chapter" data-level="26" data-path="dimplot2d.html"><a href="dimplot2d.html"><i class="fa fa-check"></i><b>26</b> dimPlot2D(</a></li>
<li class="chapter" data-level="27" data-path="g_mini.html"><a href="g_mini.html"><i class="fa fa-check"></i><b>27</b> g_mini,</a></li>
<li class="chapter" data-level="28" data-path="spat_unit-cell.html"><a href="spat_unit-cell.html"><i class="fa fa-check"></i><b>28</b> spat_unit = âcellâ,</a></li>
<li class="chapter" data-level="29" data-path="feat_type-rna.html"><a href="feat_type-rna.html"><i class="fa fa-check"></i><b>29</b> feat_type = ârnaâ,</a></li>
<li class="chapter" data-level="30" data-path="cell_color-leiden_clus.html"><a href="cell_color-leiden_clus.html"><i class="fa fa-check"></i><b>30</b> cell_color = âleiden_clusâ,</a></li>
<li class="chapter" data-level="31" data-path="dim_reduction_name-umap.html"><a href="dim_reduction_name-umap.html"><i class="fa fa-check"></i><b>31</b> dim_reduction_name = âumap.projectionâ,</a></li>
<li class="chapter" data-level="32" data-path="dim_reduction_to_use-umap.html"><a href="dim_reduction_to_use-umap.html"><i class="fa fa-check"></i><b>32</b> dim_reduction_to_use = âumapâ,</a></li>
<li class="chapter" data-level="33" data-path="save_param-list.html"><a href="save_param-list.html"><i class="fa fa-check"></i><b>33</b> save_param = list(</a></li>
<li class="chapter" data-level="34" data-path="save_name-mini_umap_1.html"><a href="save_name-mini_umap_1.html"><i class="fa fa-check"></i><b>34</b> save_name = âmini_umap_1.2â</a></li>
<li class="chapter" data-level="35" data-path="section-1.html"><a href="section-1.html"><i class="fa fa-check"></i><b>35</b> )</a></li>
<li class="chapter" data-level="36" data-path="section-2.html"><a href="section-2.html"><i class="fa fa-check"></i><b>36</b> )</a></li>
<li class="chapter" data-level="37" data-path="seems-like-a-good-res.html"><a href="seems-like-a-good-res.html"><i class="fa fa-check"></i><b>37</b> 1.4 seems like a good res</a></li>
<li class="chapter" data-level="38" data-path="project-values-back-via-knn-classifier-fnnknn.html"><a href="project-values-back-via-knn-classifier-fnnknn.html"><i class="fa fa-check"></i><b>38</b> project values back via kNN classifier (FNN::knn)</a></li>
<li class="chapter" data-level="39" data-path="spatplot2dg-cell_color-major_clus.html"><a href="spatplot2dg-cell_color-major_clus.html"><i class="fa fa-check"></i><b>39</b> spatPlot2D(g, cell_color = âmajor_clusâ,</a></li>
<li class="chapter" data-level="40" data-path="point_size-0.html"><a href="point_size-0.html"><i class="fa fa-check"></i><b>40</b> point_size = 0.1,</a></li>
<li class="chapter" data-level="41" data-path="point_shape-no_border.html"><a href="point_shape-no_border.html"><i class="fa fa-check"></i><b>41</b> point_shape = âno_borderâ,</a></li>
<li class="chapter" data-level="42" data-path="cell_color_code-clus_colorc1-1017-29c1359-14-17-11.html"><a href="cell_color_code-clus_colorc1-1017-29c1359-14-17-11.html"><i class="fa fa-check"></i><b>42</b> cell_color_code = clus_color[c(1, 10:17, 2:9)][c(1:3,5:9, 14, 17, 11)],</a></li>
<li class="chapter" data-level="43" data-path="background-black.html"><a href="background-black.html"><i class="fa fa-check"></i><b>43</b> background = âblackâ,</a></li>
<li class="chapter" data-level="44" data-path="save_param-list-1.html"><a href="save_param-list-1.html"><i class="fa fa-check"></i><b>44</b> save_param = list(</a></li>
<li class="chapter" data-level="45" data-path="base_width-15.html"><a href="base_width-15.html"><i class="fa fa-check"></i><b>45</b> base_width = 15,</a></li>
<li class="chapter" data-level="46" data-path="base_height-6.html"><a href="base_height-6.html"><i class="fa fa-check"></i><b>46</b> base_height = 6,</a></li>
<li class="chapter" data-level="47" data-path="save_name-spat_major.html"><a href="save_name-spat_major.html"><i class="fa fa-check"></i><b>47</b> save_name = âspat_majorâ</a></li>
<li class="chapter" data-level="48" data-path="section-3.html"><a href="section-3.html"><i class="fa fa-check"></i><b>48</b> ))</a></li>
<li class="chapter" data-level="49" data-path="niches.html"><a href="niches.html"><i class="fa fa-check"></i><b>49</b> niches</a></li>
<li class="chapter" data-level="50" data-path="visualize-niche-level-enrichment-for-leiden-cluster-3-in-spat_unit-cell.html"><a href="visualize-niche-level-enrichment-for-leiden-cluster-3-in-spat_unit-cell.html"><i class="fa fa-check"></i><b>50</b> visualize niche-level enrichment for leiden cluster 3 in spat_unit âcellâ</a></li>
<li class="chapter" data-level="51" data-path="these-enrichments-are-essentially-a-measure-of-how-many-cells-of-each.html"><a href="these-enrichments-are-essentially-a-measure-of-how-many-cells-of-each.html"><i class="fa fa-check"></i><b>51</b> These enrichments are essentially a measure of how many cells of each</a></li>
<li class="chapter" data-level="52" data-path="leiden-cluster-exist-in-the-local-region.html"><a href="leiden-cluster-exist-in-the-local-region.html"><i class="fa fa-check"></i><b>52</b> leiden cluster exist in the local region</a></li>
<li class="chapter" data-level="53" data-path="section-4.html"><a href="section-4.html"><i class="fa fa-check"></i><b>53</b> </a></li>
<li class="chapter" data-level="54" data-path="using-kmeans-we-can-classify-each-cell-by-its-niche-leiden-cluster-proportions.html"><a href="using-kmeans-we-can-classify-each-cell-by-its-niche-leiden-cluster-proportions.html"><i class="fa fa-check"></i><b>54</b> Using kmeans, we can classify each cell by its niche leiden cluster proportions</a></li>
<li class="chapter" data-level="55" data-path="add-kmeans-clustering-of-niches-to-cell-metadata.html"><a href="add-kmeans-clustering-of-niches-to-cell-metadata.html"><i class="fa fa-check"></i><b>55</b> add kmeans clustering of niches to cell metadata</a></li>
<li class="chapter" data-level="56" data-path="spatially-visualize-the-niches.html"><a href="spatially-visualize-the-niches.html"><i class="fa fa-check"></i><b>56</b> Spatially visualize the niches</a></li>
<li class="chapter" data-level="57" data-path="gene-enrichment.html"><a href="gene-enrichment.html"><i class="fa fa-check"></i><b>57</b> Gene enrichment</a></li>
<li class="chapter" data-level="58" data-path="violinplot.html"><a href="violinplot.html"><i class="fa fa-check"></i><b>58</b> violinplot</a></li>
<li class="chapter" data-level="59" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html"><i class="fa fa-check"></i><b>59</b> Spatial proteomics: multiplex IF</a>
<ul>
<li class="chapter" data-level="59.1" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#read-in-raw-data-2"><i class="fa fa-check"></i><b>59.1</b> Read in raw data</a></li>
<li class="chapter" data-level="59.2" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#overlap-intensity-workflows"><i class="fa fa-check"></i><b>59.2</b> Overlap intensity &amp; workflows</a></li>
</ul></li>
<li class="chapter" data-level="60" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>60</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="60.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>60.1</b> Create individual giotto objects</a></li>
<li class="chapter" data-level="60.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>60.2</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="60.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>60.3</b> Perform Harmony and default workflows</a></li>
<li class="chapter" data-level="60.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizations-1"><i class="fa fa-check"></i><b>60.4</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="61" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>61</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="61.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms"><i class="fa fa-check"></i><b>61.1</b> Spatial transforms</a>
<ul>
<li class="chapter" data-level="61.1.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-transforms-1"><i class="fa fa-check"></i><b>61.1.1</b> Spatial transforms:</a></li>
<li class="chapter" data-level="61.1.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-utilities"><i class="fa fa-check"></i><b>61.1.2</b> Spatial utilities:</a></li>
<li class="chapter" data-level="61.1.3" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#spatial-classes"><i class="fa fa-check"></i><b>61.1.3</b> Spatial classes:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="62" data-path="image-transforms.html"><a href="image-transforms.html"><i class="fa fa-check"></i><b>62</b> Image transforms</a>
<ul>
<li class="chapter" data-level="62.1" data-path="image-transforms.html"><a href="image-transforms.html#co-registration-of-datasets"><i class="fa fa-check"></i><b>62.1</b> Co-registration of datasets</a></li>
<li class="chapter" data-level="62.2" data-path="image-transforms.html"><a href="image-transforms.html#examples-in-giotto-suite-manuscript"><i class="fa fa-check"></i><b>62.2</b> Examples in giotto suite manuscript</a></li>
</ul></li>
<li class="chapter" data-level="63" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>63</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="63.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>63.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="63.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>63.2</b> Download dataset</a></li>
<li class="chapter" data-level="63.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>63.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="63.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>63.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="63.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>63.5</b> RNA processing</a></li>
<li class="chapter" data-level="63.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>63.6</b> Protein processing</a></li>
<li class="chapter" data-level="63.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>63.7</b> Multi-omics integration</a></li>
<li class="chapter" data-level="63.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>63.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="64" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>64</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="64.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#load-giotto-object"><i class="fa fa-check"></i><b>64.1</b> Load Giotto object</a></li>
<li class="chapter" data-level="64.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>64.2</b> SpatialExperiment</a>
<ul>
<li class="chapter" data-level="64.2.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#convert-giotto-object-to-spatialexperiment-object"><i class="fa fa-check"></i><b>64.2.1</b> Convert Giotto Object to SpatialExperiment Object</a></li>
</ul></li>
<li class="chapter" data-level="64.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#identify-top-spatially-variable-genes-with-nnsvg"><i class="fa fa-check"></i><b>64.3</b> Identify top spatially variable genes with nnSVG</a>
<ul>
<li class="chapter" data-level="64.3.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#conversion-of-spatialexperiment-object-back-to-giotto"><i class="fa fa-check"></i><b>64.3.1</b> Conversion of SpatialExperiment object back to Giotto</a></li>
<li class="chapter" data-level="64.3.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#plotting-top-genes-from-nnsvg-in-giotto"><i class="fa fa-check"></i><b>64.3.2</b> Plotting top genes from nnSVG in Giotto</a></li>
</ul></li>
<li class="chapter" data-level="64.4" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>64.4</b> Seurat</a>
<ul>
<li class="chapter" data-level="64.4.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#giotto-to-seurat"><i class="fa fa-check"></i><b>64.4.1</b> Giotto to Seurat</a></li>
</ul></li>
<li class="chapter" data-level="64.5" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndataspatialdata"><i class="fa fa-check"></i><b>64.5</b> AnnData/SpatialData</a></li>
</ul></li>
<li class="chapter" data-level="65" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>65</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="65.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis"><i class="fa fa-check"></i><b>65.1</b> Spatial niche trajectory analysis</a>
<ul>
<li class="chapter" data-level="65.1.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#prepare"><i class="fa fa-check"></i><b>65.1.1</b> Prepare</a></li>
<li class="chapter" data-level="65.1.2" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#running"><i class="fa fa-check"></i><b>65.1.2</b> Running</a></li>
<li class="chapter" data-level="65.1.3" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#visualization"><i class="fa fa-check"></i><b>65.1.3</b> Visualization</a></li>
<li class="chapter" data-level="65.1.4" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#section-5"><i class="fa fa-check"></i><b>65.1.4</b> </a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="66" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>66</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="66.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>66.1</b> Kriging</a></li>
</ul></li>
<li class="chapter" data-level="67" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>67</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="67.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>67.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="multi-omics-integration" class="section level1 hasAnchor" number="63">
<h1><span class="header-section-number">63</span> Multi-omics integration<a href="multi-omics-integration.html#multi-omics-integration" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Joselyn Cristina ChÃ¡vez Fuentes</p>
<p>August 7th 2024</p>
<div id="introduction-to-the-spatial-dataset-1" class="section level2 hasAnchor" number="63.1">
<h2><span class="header-section-number">63.1</span> Introduction to the spatial dataset<a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <a href="https://www.10xgenomics.com/resources/datasets/gene-and-protein-expression-library-of-human-glioblastoma-cytassist-ffpe-2-standard/">Human glioblastoma (FFPE) dataset</a> was obtained from 10X Genomics. The tissue was sectioned as described in visium_glioblastoma CytAssist Spatial Gene Expression for FFPE â Tissue Preparation Guide Demonstrated Protocol (CG000518). 5 Âµm tissue sections were placed on Superfrost glass slides, then IF stained following deparaffinization, then hard coverslipped. Sections were imaged, decoverslipped, followed by Demonstrated Protocol (CG000494).</p>
<p>More information about this dataset can be found <a href="https://www.10xgenomics.com/resources/datasets/gene-and-protein-expression-library-of-human-glioblastoma-cytassist-ffpe-2-standard">here</a>.</p>
</div>
<div id="download-dataset-1" class="section level2 hasAnchor" number="63.2">
<h2><span class="header-section-number">63.2</span> Download dataset<a href="multi-omics-integration.html#download-dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You need to download the expression matrix and spatial information by running these commands:</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="multi-omics-integration.html#cb110-1" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="st">&quot;data&quot;</span>)</span>
<span id="cb110-2"><a href="multi-omics-integration.html#cb110-2" tabindex="-1"></a></span>
<span id="cb110-3"><a href="multi-omics-integration.html#cb110-3" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://cf.10xgenomics.com/samples/spatial-exp/2.1.0/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma_raw_feature_bc_matrix.tar.gz&quot;</span>,</span>
<span id="cb110-4"><a href="multi-omics-integration.html#cb110-4" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="st">&quot;data/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma_raw_feature_bc_matrix.tar.gz&quot;</span>)</span>
<span id="cb110-5"><a href="multi-omics-integration.html#cb110-5" tabindex="-1"></a></span>
<span id="cb110-6"><a href="multi-omics-integration.html#cb110-6" tabindex="-1"></a><span class="fu">download.file</span>(<span class="at">url =</span> <span class="st">&quot;https://cf.10xgenomics.com/samples/spatial-exp/2.1.0/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma_spatial.tar.gz&quot;</span>,</span>
<span id="cb110-7"><a href="multi-omics-integration.html#cb110-7" tabindex="-1"></a>              <span class="at">destfile =</span> <span class="st">&quot;data/CytAssist_FFPE_Protein_Expression_Human_Glioblastoma_spatial.tar.gz&quot;</span>)</span></code></pre></div>
<p>After downloading, unzip the gz files. You should get the âraw_feature_bc_matrixâ and âspatialâ folders inside âdata/â.</p>
</div>
<div id="create-the-giotto-object-1" class="section level2 hasAnchor" number="63.3">
<h2><span class="header-section-number">63.3</span> Create the Giotto object<a href="multi-omics-integration.html#create-the-giotto-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The minimum requirements are:</p>
<ul>
<li>matrix with expression information (or the path to)</li>
<li>x,y(,z) coordinates for cells or spots (or the path to)</li>
</ul>
<p>createGiottoVisiumObject() will automatically detect both RNA and Protein modalities in the expression matrix and will create a multi-omics Giotto object.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="multi-omics-integration.html#cb111-1" tabindex="-1"></a><span class="fu">library</span>(Giotto)</span>
<span id="cb111-2"><a href="multi-omics-integration.html#cb111-2" tabindex="-1"></a></span>
<span id="cb111-3"><a href="multi-omics-integration.html#cb111-3" tabindex="-1"></a><span class="do">## Set instructions</span></span>
<span id="cb111-4"><a href="multi-omics-integration.html#cb111-4" tabindex="-1"></a>results_folder <span class="ot">&lt;-</span> <span class="st">&quot;results/&quot;</span></span>
<span id="cb111-5"><a href="multi-omics-integration.html#cb111-5" tabindex="-1"></a></span>
<span id="cb111-6"><a href="multi-omics-integration.html#cb111-6" tabindex="-1"></a>python_path <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb111-7"><a href="multi-omics-integration.html#cb111-7" tabindex="-1"></a></span>
<span id="cb111-8"><a href="multi-omics-integration.html#cb111-8" tabindex="-1"></a>instructions <span class="ot">&lt;-</span> <span class="fu">createGiottoInstructions</span>(</span>
<span id="cb111-9"><a href="multi-omics-integration.html#cb111-9" tabindex="-1"></a>    <span class="at">save_dir =</span> results_folder,</span>
<span id="cb111-10"><a href="multi-omics-integration.html#cb111-10" tabindex="-1"></a>    <span class="at">save_plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb111-11"><a href="multi-omics-integration.html#cb111-11" tabindex="-1"></a>    <span class="at">show_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb111-12"><a href="multi-omics-integration.html#cb111-12" tabindex="-1"></a>    <span class="at">return_plot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb111-13"><a href="multi-omics-integration.html#cb111-13" tabindex="-1"></a>    <span class="at">python_path =</span> python_path</span>
<span id="cb111-14"><a href="multi-omics-integration.html#cb111-14" tabindex="-1"></a>)</span>
<span id="cb111-15"><a href="multi-omics-integration.html#cb111-15" tabindex="-1"></a></span>
<span id="cb111-16"><a href="multi-omics-integration.html#cb111-16" tabindex="-1"></a><span class="co"># Provide the path to the visium_glioblastoma folder</span></span>
<span id="cb111-17"><a href="multi-omics-integration.html#cb111-17" tabindex="-1"></a>data_path <span class="ot">&lt;-</span> <span class="st">&quot;data&quot;</span></span>
<span id="cb111-18"><a href="multi-omics-integration.html#cb111-18" tabindex="-1"></a></span>
<span id="cb111-19"><a href="multi-omics-integration.html#cb111-19" tabindex="-1"></a><span class="co"># Create object directly from the visium_glioblastoma folder</span></span>
<span id="cb111-20"><a href="multi-omics-integration.html#cb111-20" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">createGiottoVisiumObject</span>(</span>
<span id="cb111-21"><a href="multi-omics-integration.html#cb111-21" tabindex="-1"></a>    <span class="at">visium_dir =</span> data_path,</span>
<span id="cb111-22"><a href="multi-omics-integration.html#cb111-22" tabindex="-1"></a>    <span class="at">expr_data =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb111-23"><a href="multi-omics-integration.html#cb111-23" tabindex="-1"></a>    <span class="at">png_name =</span> <span class="st">&quot;tissue_lowres_image.png&quot;</span>,</span>
<span id="cb111-24"><a href="multi-omics-integration.html#cb111-24" tabindex="-1"></a>    <span class="at">gene_column_index =</span> <span class="dv">2</span>,</span>
<span id="cb111-25"><a href="multi-omics-integration.html#cb111-25" tabindex="-1"></a>    <span class="at">instructions =</span> instructions</span>
<span id="cb111-26"><a href="multi-omics-integration.html#cb111-26" tabindex="-1"></a>)</span></code></pre></div>
<ul>
<li>Print the information of the object, note that both rna and protein are listed in the expression slot</li>
</ul>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="multi-omics-integration.html#cb112-1" tabindex="-1"></a>visium_glioblastoma</span></code></pre></div>
</div>
<div id="subset-on-spots-that-were-covered-by-tissue-1" class="section level2 hasAnchor" number="63.4">
<h2><span class="header-section-number">63.4</span> Subset on spots that were covered by tissue<a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="multi-omics-integration.html#cb113-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(</span>
<span id="cb113-2"><a href="multi-omics-integration.html#cb113-2" tabindex="-1"></a>    <span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb113-3"><a href="multi-omics-integration.html#cb113-3" tabindex="-1"></a>    <span class="at">cell_color =</span> <span class="st">&quot;in_tissue&quot;</span>,</span>
<span id="cb113-4"><a href="multi-omics-integration.html#cb113-4" tabindex="-1"></a>    <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb113-5"><a href="multi-omics-integration.html#cb113-5" tabindex="-1"></a>    <span class="at">cell_color_code =</span> <span class="fu">c</span>(<span class="st">&quot;0&quot;</span> <span class="ot">=</span> <span class="st">&quot;lightgrey&quot;</span>, <span class="st">&quot;1&quot;</span> <span class="ot">=</span> <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb113-6"><a href="multi-omics-integration.html#cb113-6" tabindex="-1"></a>    <span class="at">show_image =</span> <span class="cn">TRUE</span>,</span>
<span id="cb113-7"><a href="multi-omics-integration.html#cb113-7" tabindex="-1"></a>    <span class="at">image_name =</span> <span class="st">&quot;image&quot;</span></span>
<span id="cb113-8"><a href="multi-omics-integration.html#cb113-8" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="img/03_session3/0-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="multi-omics-integration.html#cb114-1" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">getCellMetadata</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb114-2"><a href="multi-omics-integration.html#cb114-2" tabindex="-1"></a>                            <span class="at">output =</span> <span class="st">&quot;data.table&quot;</span>)</span>
<span id="cb114-3"><a href="multi-omics-integration.html#cb114-3" tabindex="-1"></a></span>
<span id="cb114-4"><a href="multi-omics-integration.html#cb114-4" tabindex="-1"></a>in_tissue_barcodes <span class="ot">&lt;-</span> metadata[in_tissue <span class="sc">==</span> <span class="dv">1</span>]<span class="sc">$</span>cell_ID</span>
<span id="cb114-5"><a href="multi-omics-integration.html#cb114-5" tabindex="-1"></a></span>
<span id="cb114-6"><a href="multi-omics-integration.html#cb114-6" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">subsetGiotto</span>(visium_glioblastoma, </span>
<span id="cb114-7"><a href="multi-omics-integration.html#cb114-7" tabindex="-1"></a>                                    <span class="at">cell_ids =</span> in_tissue_barcodes)</span></code></pre></div>
</div>
<div id="rna-processing" class="section level2 hasAnchor" number="63.5">
<h2><span class="header-section-number">63.5</span> RNA processing<a href="multi-omics-integration.html#rna-processing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Filtering, normalization, and statistics</li>
</ul>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="multi-omics-integration.html#cb115-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(</span>
<span id="cb115-2"><a href="multi-omics-integration.html#cb115-2" tabindex="-1"></a>    <span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb115-3"><a href="multi-omics-integration.html#cb115-3" tabindex="-1"></a>    <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb115-4"><a href="multi-omics-integration.html#cb115-4" tabindex="-1"></a>    <span class="at">feat_det_in_min_cells =</span> <span class="dv">50</span>,</span>
<span id="cb115-5"><a href="multi-omics-integration.html#cb115-5" tabindex="-1"></a>    <span class="at">min_det_feats_per_cell =</span> <span class="dv">1000</span>,</span>
<span id="cb115-6"><a href="multi-omics-integration.html#cb115-6" tabindex="-1"></a>    <span class="at">expression_values =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb115-7"><a href="multi-omics-integration.html#cb115-7" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb115-8"><a href="multi-omics-integration.html#cb115-8" tabindex="-1"></a></span>
<span id="cb115-9"><a href="multi-omics-integration.html#cb115-9" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb115-10"><a href="multi-omics-integration.html#cb115-10" tabindex="-1"></a>                                       <span class="at">scalefactor =</span> <span class="dv">6000</span>,</span>
<span id="cb115-11"><a href="multi-omics-integration.html#cb115-11" tabindex="-1"></a>                                       <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb115-12"><a href="multi-omics-integration.html#cb115-12" tabindex="-1"></a></span>
<span id="cb115-13"><a href="multi-omics-integration.html#cb115-13" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(<span class="at">gobject =</span> visium_glioblastoma)</span></code></pre></div>
<ul>
<li>Dimension reduction</li>
</ul>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="multi-omics-integration.html#cb116-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">calculateHVF</span>(<span class="at">gobject =</span> visium_glioblastoma)</span>
<span id="cb116-2"><a href="multi-omics-integration.html#cb116-2" tabindex="-1"></a></span>
<span id="cb116-3"><a href="multi-omics-integration.html#cb116-3" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_glioblastoma)</span></code></pre></div>
<ul>
<li>Clustering</li>
</ul>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="multi-omics-integration.html#cb117-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visium_glioblastoma,</span>
<span id="cb117-2"><a href="multi-omics-integration.html#cb117-2" tabindex="-1"></a>                               <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb117-3"><a href="multi-omics-integration.html#cb117-3" tabindex="-1"></a></span>
<span id="cb117-4"><a href="multi-omics-integration.html#cb117-4" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb117-5"><a href="multi-omics-integration.html#cb117-5" tabindex="-1"></a>                                            <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb117-6"><a href="multi-omics-integration.html#cb117-6" tabindex="-1"></a>                                            <span class="at">k =</span> <span class="dv">30</span>)</span>
<span id="cb117-7"><a href="multi-omics-integration.html#cb117-7" tabindex="-1"></a></span>
<span id="cb117-8"><a href="multi-omics-integration.html#cb117-8" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb117-9"><a href="multi-omics-integration.html#cb117-9" tabindex="-1"></a>                                       <span class="at">resolution =</span> <span class="dv">1</span>,</span>
<span id="cb117-10"><a href="multi-omics-integration.html#cb117-10" tabindex="-1"></a>                                       <span class="at">n_iterations =</span> <span class="dv">1000</span>)</span></code></pre></div>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="multi-omics-integration.html#cb118-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb118-2"><a href="multi-omics-integration.html#cb118-2" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb118-3"><a href="multi-omics-integration.html#cb118-3" tabindex="-1"></a>         <span class="at">show_NN_network =</span> <span class="cn">TRUE</span>,</span>
<span id="cb118-4"><a href="multi-omics-integration.html#cb118-4" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/03_session3/1-UMAP.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="multi-omics-integration.html#cb119-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb119-2"><a href="multi-omics-integration.html#cb119-2" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb119-3"><a href="multi-omics-integration.html#cb119-3" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb119-4"><a href="multi-omics-integration.html#cb119-4" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/03_session3/2-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="protein-processing" class="section level2 hasAnchor" number="63.6">
<h2><span class="header-section-number">63.6</span> Protein processing<a href="multi-omics-integration.html#protein-processing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Filtering, normalization, and statistics</li>
</ul>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="multi-omics-integration.html#cb120-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">filterGiotto</span>(<span class="at">gobject =</span> visium_glioblastoma, </span>
<span id="cb120-2"><a href="multi-omics-integration.html#cb120-2" tabindex="-1"></a>                                    <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb120-3"><a href="multi-omics-integration.html#cb120-3" tabindex="-1"></a>                                    <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb120-4"><a href="multi-omics-integration.html#cb120-4" tabindex="-1"></a>                                    <span class="at">expression_threshold =</span> <span class="dv">1</span>,</span>
<span id="cb120-5"><a href="multi-omics-integration.html#cb120-5" tabindex="-1"></a>                                    <span class="at">feat_det_in_min_cells =</span> <span class="dv">50</span>, </span>
<span id="cb120-6"><a href="multi-omics-integration.html#cb120-6" tabindex="-1"></a>                                    <span class="at">min_det_feats_per_cell =</span> <span class="dv">1</span>,</span>
<span id="cb120-7"><a href="multi-omics-integration.html#cb120-7" tabindex="-1"></a>                                    <span class="at">expression_values =</span> <span class="st">&quot;raw&quot;</span>, </span>
<span id="cb120-8"><a href="multi-omics-integration.html#cb120-8" tabindex="-1"></a>                                    <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-9"><a href="multi-omics-integration.html#cb120-9" tabindex="-1"></a></span>
<span id="cb120-10"><a href="multi-omics-integration.html#cb120-10" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">normalizeGiotto</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb120-11"><a href="multi-omics-integration.html#cb120-11" tabindex="-1"></a>                                       <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb120-12"><a href="multi-omics-integration.html#cb120-12" tabindex="-1"></a>                                       <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>, </span>
<span id="cb120-13"><a href="multi-omics-integration.html#cb120-13" tabindex="-1"></a>                                       <span class="at">scalefactor =</span> <span class="dv">6000</span>, </span>
<span id="cb120-14"><a href="multi-omics-integration.html#cb120-14" tabindex="-1"></a>                                       <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb120-15"><a href="multi-omics-integration.html#cb120-15" tabindex="-1"></a></span>
<span id="cb120-16"><a href="multi-omics-integration.html#cb120-16" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">addStatistics</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb120-17"><a href="multi-omics-integration.html#cb120-17" tabindex="-1"></a>                                     <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb120-18"><a href="multi-omics-integration.html#cb120-18" tabindex="-1"></a>                                     <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<ul>
<li>Dimension reduction</li>
</ul>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="multi-omics-integration.html#cb121-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runPCA</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb121-2"><a href="multi-omics-integration.html#cb121-2" tabindex="-1"></a>                              <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb121-3"><a href="multi-omics-integration.html#cb121-3" tabindex="-1"></a>                              <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>)</span></code></pre></div>
<ul>
<li>Clustering</li>
</ul>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="multi-omics-integration.html#cb122-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runUMAP</span>(visium_glioblastoma,</span>
<span id="cb122-2"><a href="multi-omics-integration.html#cb122-2" tabindex="-1"></a>                               <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb122-3"><a href="multi-omics-integration.html#cb122-3" tabindex="-1"></a>                               <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>, </span>
<span id="cb122-4"><a href="multi-omics-integration.html#cb122-4" tabindex="-1"></a>                               <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>)</span>
<span id="cb122-5"><a href="multi-omics-integration.html#cb122-5" tabindex="-1"></a></span>
<span id="cb122-6"><a href="multi-omics-integration.html#cb122-6" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb122-7"><a href="multi-omics-integration.html#cb122-7" tabindex="-1"></a>                                            <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb122-8"><a href="multi-omics-integration.html#cb122-8" tabindex="-1"></a>                                            <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb122-9"><a href="multi-omics-integration.html#cb122-9" tabindex="-1"></a>                                            <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb122-10"><a href="multi-omics-integration.html#cb122-10" tabindex="-1"></a>                                            <span class="at">k =</span> <span class="dv">30</span>)</span>
<span id="cb122-11"><a href="multi-omics-integration.html#cb122-11" tabindex="-1"></a></span>
<span id="cb122-12"><a href="multi-omics-integration.html#cb122-12" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb122-13"><a href="multi-omics-integration.html#cb122-13" tabindex="-1"></a>                                       <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb122-14"><a href="multi-omics-integration.html#cb122-14" tabindex="-1"></a>                                       <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb122-15"><a href="multi-omics-integration.html#cb122-15" tabindex="-1"></a>                                       <span class="at">resolution =</span> <span class="dv">1</span>,</span>
<span id="cb122-16"><a href="multi-omics-integration.html#cb122-16" tabindex="-1"></a>                                       <span class="at">n_iterations =</span> <span class="dv">1000</span>)</span></code></pre></div>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="multi-omics-integration.html#cb123-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb123-2"><a href="multi-omics-integration.html#cb123-2" tabindex="-1"></a>         <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>, </span>
<span id="cb123-3"><a href="multi-omics-integration.html#cb123-3" tabindex="-1"></a>         <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>, </span>
<span id="cb123-4"><a href="multi-omics-integration.html#cb123-4" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>, </span>
<span id="cb123-5"><a href="multi-omics-integration.html#cb123-5" tabindex="-1"></a>         <span class="at">show_NN_network =</span> <span class="cn">TRUE</span>, </span>
<span id="cb123-6"><a href="multi-omics-integration.html#cb123-6" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/03_session3/3-UMAP.png" width="80%" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="multi-omics-integration.html#cb124-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb124-2"><a href="multi-omics-integration.html#cb124-2" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb124-3"><a href="multi-omics-integration.html#cb124-3" tabindex="-1"></a>           <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb124-4"><a href="multi-omics-integration.html#cb124-4" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb124-5"><a href="multi-omics-integration.html#cb124-5" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;leiden_clus&quot;</span>,</span>
<span id="cb124-6"><a href="multi-omics-integration.html#cb124-6" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="img/03_session3/4-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="multi-omics-integration-1" class="section level2 hasAnchor" number="63.7">
<h2><span class="header-section-number">63.7</span> Multi-omics integration<a href="multi-omics-integration.html#multi-omics-integration-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Calculate kNN</li>
</ul>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="multi-omics-integration.html#cb125-1" tabindex="-1"></a><span class="do">## RNA modality</span></span>
<span id="cb125-2"><a href="multi-omics-integration.html#cb125-2" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb125-3"><a href="multi-omics-integration.html#cb125-3" tabindex="-1"></a>                                            <span class="at">type =</span> <span class="st">&quot;kNN&quot;</span>,</span>
<span id="cb125-4"><a href="multi-omics-integration.html#cb125-4" tabindex="-1"></a>                                            <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb125-5"><a href="multi-omics-integration.html#cb125-5" tabindex="-1"></a>                                            <span class="at">k =</span> <span class="dv">20</span>)</span>
<span id="cb125-6"><a href="multi-omics-integration.html#cb125-6" tabindex="-1"></a></span>
<span id="cb125-7"><a href="multi-omics-integration.html#cb125-7" tabindex="-1"></a><span class="do">## Protein modality</span></span>
<span id="cb125-8"><a href="multi-omics-integration.html#cb125-8" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">createNearestNetwork</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb125-9"><a href="multi-omics-integration.html#cb125-9" tabindex="-1"></a>                                            <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb125-10"><a href="multi-omics-integration.html#cb125-10" tabindex="-1"></a>                                            <span class="at">feat_type =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb125-11"><a href="multi-omics-integration.html#cb125-11" tabindex="-1"></a>                                            <span class="at">type =</span> <span class="st">&quot;kNN&quot;</span>,</span>
<span id="cb125-12"><a href="multi-omics-integration.html#cb125-12" tabindex="-1"></a>                                            <span class="at">dimensions_to_use =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,</span>
<span id="cb125-13"><a href="multi-omics-integration.html#cb125-13" tabindex="-1"></a>                                            <span class="at">k =</span> <span class="dv">20</span>)</span></code></pre></div>
<ul>
<li>Run WNN</li>
</ul>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="multi-omics-integration.html#cb126-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runWNN</span>(visium_glioblastoma,</span>
<span id="cb126-2"><a href="multi-omics-integration.html#cb126-2" tabindex="-1"></a>                              <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb126-3"><a href="multi-omics-integration.html#cb126-3" tabindex="-1"></a>                              <span class="at">modality_1 =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb126-4"><a href="multi-omics-integration.html#cb126-4" tabindex="-1"></a>                              <span class="at">modality_2 =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb126-5"><a href="multi-omics-integration.html#cb126-5" tabindex="-1"></a>                              <span class="at">pca_name_modality_1 =</span> <span class="st">&quot;pca&quot;</span>,</span>
<span id="cb126-6"><a href="multi-omics-integration.html#cb126-6" tabindex="-1"></a>                              <span class="at">pca_name_modality_2 =</span> <span class="st">&quot;protein.pca&quot;</span>,</span>
<span id="cb126-7"><a href="multi-omics-integration.html#cb126-7" tabindex="-1"></a>                              <span class="at">k =</span> <span class="dv">20</span>,</span>
<span id="cb126-8"><a href="multi-omics-integration.html#cb126-8" tabindex="-1"></a>                              <span class="at">integrated_feat_type =</span> <span class="cn">NULL</span>,</span>
<span id="cb126-9"><a href="multi-omics-integration.html#cb126-9" tabindex="-1"></a>                              <span class="at">matrix_result_name =</span> <span class="cn">NULL</span>,</span>
<span id="cb126-10"><a href="multi-omics-integration.html#cb126-10" tabindex="-1"></a>                              <span class="at">w_name_modality_1 =</span> <span class="cn">NULL</span>,</span>
<span id="cb126-11"><a href="multi-omics-integration.html#cb126-11" tabindex="-1"></a>                              <span class="at">w_name_modality_2 =</span> <span class="cn">NULL</span>,</span>
<span id="cb126-12"><a href="multi-omics-integration.html#cb126-12" tabindex="-1"></a>                              <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<ul>
<li>Run Integrated umap</li>
</ul>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="multi-omics-integration.html#cb127-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">runIntegratedUMAP</span>(visium_glioblastoma,</span>
<span id="cb127-2"><a href="multi-omics-integration.html#cb127-2" tabindex="-1"></a>                                         <span class="at">modality1 =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb127-3"><a href="multi-omics-integration.html#cb127-3" tabindex="-1"></a>                                         <span class="at">modality2 =</span> <span class="st">&quot;protein&quot;</span>,</span>
<span id="cb127-4"><a href="multi-omics-integration.html#cb127-4" tabindex="-1"></a>                                         <span class="at">spread =</span> <span class="dv">5</span>,</span>
<span id="cb127-5"><a href="multi-omics-integration.html#cb127-5" tabindex="-1"></a>                                         <span class="at">min_dist =</span> <span class="fl">0.5</span>,</span>
<span id="cb127-6"><a href="multi-omics-integration.html#cb127-6" tabindex="-1"></a>                                         <span class="at">force =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<ul>
<li>Calculate integrated clusters</li>
</ul>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="multi-omics-integration.html#cb128-1" tabindex="-1"></a>visium_glioblastoma <span class="ot">&lt;-</span> <span class="fu">doLeidenCluster</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb128-2"><a href="multi-omics-integration.html#cb128-2" tabindex="-1"></a>                                       <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb128-3"><a href="multi-omics-integration.html#cb128-3" tabindex="-1"></a>                                       <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb128-4"><a href="multi-omics-integration.html#cb128-4" tabindex="-1"></a>                                       <span class="at">nn_network_to_use =</span> <span class="st">&quot;kNN&quot;</span>,</span>
<span id="cb128-5"><a href="multi-omics-integration.html#cb128-5" tabindex="-1"></a>                                       <span class="at">network_name =</span> <span class="st">&quot;integrated_kNN&quot;</span>,</span>
<span id="cb128-6"><a href="multi-omics-integration.html#cb128-6" tabindex="-1"></a>                                       <span class="at">name =</span> <span class="st">&quot;integrated_leiden_clus&quot;</span>,</span>
<span id="cb128-7"><a href="multi-omics-integration.html#cb128-7" tabindex="-1"></a>                                       <span class="at">resolution =</span> <span class="dv">1</span>)</span></code></pre></div>
<ul>
<li>Visualize the integrated umap</li>
</ul>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="multi-omics-integration.html#cb129-1" tabindex="-1"></a><span class="fu">plotUMAP</span>(<span class="at">gobject =</span> visium_glioblastoma,</span>
<span id="cb129-2"><a href="multi-omics-integration.html#cb129-2" tabindex="-1"></a>         <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb129-3"><a href="multi-omics-integration.html#cb129-3" tabindex="-1"></a>         <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb129-4"><a href="multi-omics-integration.html#cb129-4" tabindex="-1"></a>         <span class="at">cell_color =</span> <span class="st">&quot;integrated_leiden_clus&quot;</span>,</span>
<span id="cb129-5"><a href="multi-omics-integration.html#cb129-5" tabindex="-1"></a>         <span class="at">dim_reduction_name =</span> <span class="st">&quot;integrated.umap&quot;</span>,</span>
<span id="cb129-6"><a href="multi-omics-integration.html#cb129-6" tabindex="-1"></a>         <span class="at">point_size =</span> <span class="fl">1.5</span>,</span>
<span id="cb129-7"><a href="multi-omics-integration.html#cb129-7" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">&quot;Integrated UMAP using Integrated Leiden clusters&quot;</span>,</span>
<span id="cb129-8"><a href="multi-omics-integration.html#cb129-8" tabindex="-1"></a>         <span class="at">axis_title =</span> <span class="dv">12</span>,</span>
<span id="cb129-9"><a href="multi-omics-integration.html#cb129-9" tabindex="-1"></a>         <span class="at">axis_text =</span> <span class="dv">10</span> )</span></code></pre></div>
<p><img src="img/03_session3/5-UMAP.png" width="80%" style="display: block; margin: auto;" /></p>
<ul>
<li>Visualize spatial plot with integrated clusters</li>
</ul>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="multi-omics-integration.html#cb130-1" tabindex="-1"></a><span class="fu">spatPlot2D</span>(visium_glioblastoma,</span>
<span id="cb130-2"><a href="multi-omics-integration.html#cb130-2" tabindex="-1"></a>           <span class="at">spat_unit =</span> <span class="st">&quot;cell&quot;</span>,</span>
<span id="cb130-3"><a href="multi-omics-integration.html#cb130-3" tabindex="-1"></a>           <span class="at">feat_type =</span> <span class="st">&quot;rna&quot;</span>,</span>
<span id="cb130-4"><a href="multi-omics-integration.html#cb130-4" tabindex="-1"></a>           <span class="at">cell_color =</span> <span class="st">&quot;integrated_leiden_clus&quot;</span>,</span>
<span id="cb130-5"><a href="multi-omics-integration.html#cb130-5" tabindex="-1"></a>           <span class="at">point_size =</span> <span class="dv">2</span>,</span>
<span id="cb130-6"><a href="multi-omics-integration.html#cb130-6" tabindex="-1"></a>           <span class="at">show_image =</span> <span class="cn">FALSE</span>,</span>
<span id="cb130-7"><a href="multi-omics-integration.html#cb130-7" tabindex="-1"></a>           <span class="at">title =</span> <span class="st">&quot;Integrated Leiden clustering&quot;</span>)</span></code></pre></div>
<p><img src="img/03_session3/6-spatPlot2D.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
<div id="session-info-2" class="section level2 hasAnchor" number="63.8">
<h2><span class="header-section-number">63.8</span> Session info<a href="multi-omics-integration.html#session-info-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="multi-omics-integration.html#cb131-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="multi-omics-integration.html#cb132-1" tabindex="-1"></a>R version <span class="dv">4</span>.<span class="fl">4.1</span> (<span class="dv">2024-06-14</span>)</span>
<span id="cb132-2"><a href="multi-omics-integration.html#cb132-2" tabindex="-1"></a>Platform<span class="sc">:</span> aarch64<span class="sc">-</span>apple<span class="sc">-</span>darwin20</span>
<span id="cb132-3"><a href="multi-omics-integration.html#cb132-3" tabindex="-1"></a>Running under<span class="sc">:</span> macOS Sonoma <span class="fl">14.5</span></span>
<span id="cb132-4"><a href="multi-omics-integration.html#cb132-4" tabindex="-1"></a></span>
<span id="cb132-5"><a href="multi-omics-integration.html#cb132-5" tabindex="-1"></a>Matrix products<span class="sc">:</span> default</span>
<span id="cb132-6"><a href="multi-omics-integration.html#cb132-6" tabindex="-1"></a>BLAS<span class="sc">:</span>   <span class="er">/</span>System<span class="sc">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>Accelerate.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>Frameworks<span class="sc">/</span>vecLib.framework<span class="sc">/</span>Versions<span class="sc">/</span>A<span class="sc">/</span>libBLAS.dylib </span>
<span id="cb132-7"><a href="multi-omics-integration.html#cb132-7" tabindex="-1"></a>LAPACK<span class="sc">:</span> <span class="er">/</span>Library<span class="sc">/</span>Frameworks<span class="sc">/</span>R.framework<span class="sc">/</span>Versions<span class="sc">/</span><span class="fl">4.4</span><span class="sc">-</span>arm64<span class="sc">/</span>Resources<span class="sc">/</span>lib<span class="sc">/</span>libRlapack.dylib;  LAPACK version <span class="dv">3</span>.<span class="fl">12.0</span></span>
<span id="cb132-8"><a href="multi-omics-integration.html#cb132-8" tabindex="-1"></a></span>
<span id="cb132-9"><a href="multi-omics-integration.html#cb132-9" tabindex="-1"></a>locale<span class="sc">:</span></span>
<span id="cb132-10"><a href="multi-omics-integration.html#cb132-10" tabindex="-1"></a>[<span class="dv">1</span>] en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>C<span class="sc">/</span>en_US.UTF<span class="dv">-8</span><span class="sc">/</span>en_US.UTF<span class="dv">-8</span></span>
<span id="cb132-11"><a href="multi-omics-integration.html#cb132-11" tabindex="-1"></a></span>
<span id="cb132-12"><a href="multi-omics-integration.html#cb132-12" tabindex="-1"></a>time zone<span class="sc">:</span> America<span class="sc">/</span>New_York</span>
<span id="cb132-13"><a href="multi-omics-integration.html#cb132-13" tabindex="-1"></a>tzcode source<span class="sc">:</span> internal</span>
<span id="cb132-14"><a href="multi-omics-integration.html#cb132-14" tabindex="-1"></a></span>
<span id="cb132-15"><a href="multi-omics-integration.html#cb132-15" tabindex="-1"></a>attached base packages<span class="sc">:</span></span>
<span id="cb132-16"><a href="multi-omics-integration.html#cb132-16" tabindex="-1"></a>[<span class="dv">1</span>] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span id="cb132-17"><a href="multi-omics-integration.html#cb132-17" tabindex="-1"></a></span>
<span id="cb132-18"><a href="multi-omics-integration.html#cb132-18" tabindex="-1"></a>other attached packages<span class="sc">:</span></span>
<span id="cb132-19"><a href="multi-omics-integration.html#cb132-19" tabindex="-1"></a>[<span class="dv">1</span>] Giotto_4.<span class="fl">0.8</span>      GiottoClass_0.<span class="fl">3.1</span></span>
<span id="cb132-20"><a href="multi-omics-integration.html#cb132-20" tabindex="-1"></a></span>
<span id="cb132-21"><a href="multi-omics-integration.html#cb132-21" tabindex="-1"></a>loaded via a <span class="fu">namespace</span> (and not attached)<span class="sc">:</span></span>
<span id="cb132-22"><a href="multi-omics-integration.html#cb132-22" tabindex="-1"></a>  [<span class="dv">1</span>] colorRamp2_0.<span class="fl">1.0</span>      deldir_2<span class="fl">.0</span><span class="dv">-4</span>          rlang_1.<span class="fl">1.4</span>          </span>
<span id="cb132-23"><a href="multi-omics-integration.html#cb132-23" tabindex="-1"></a>  [<span class="dv">4</span>] magrittr_2.<span class="fl">0.3</span>        RcppAnnoy_0.<span class="fl">0.22</span>      GiottoUtils_0.<span class="fl">1.8</span>    </span>
<span id="cb132-24"><a href="multi-omics-integration.html#cb132-24" tabindex="-1"></a>  [<span class="dv">7</span>] matrixStats_1.<span class="fl">3.0</span>     compiler_4.<span class="fl">4.1</span>        png_0<span class="fl">.1</span><span class="dv">-8</span>            </span>
<span id="cb132-25"><a href="multi-omics-integration.html#cb132-25" tabindex="-1"></a> [<span class="dv">10</span>] systemfonts_1.<span class="fl">1.0</span>     vctrs_0.<span class="fl">6.5</span>           reshape2_1.<span class="fl">4.4</span>       </span>
<span id="cb132-26"><a href="multi-omics-integration.html#cb132-26" tabindex="-1"></a> [<span class="dv">13</span>] stringr_1.<span class="fl">5.1</span>         pkgconfig_2.<span class="fl">0.3</span>       crayon_1.<span class="fl">5.3</span>         </span>
<span id="cb132-27"><a href="multi-omics-integration.html#cb132-27" tabindex="-1"></a> [<span class="dv">16</span>] fastmap_1.<span class="fl">2.0</span>         backports_1.<span class="fl">5.0</span>       magick_2.<span class="fl">8.4</span>         </span>
<span id="cb132-28"><a href="multi-omics-integration.html#cb132-28" tabindex="-1"></a> [<span class="dv">19</span>] XVector_0.<span class="fl">44.0</span>        labeling_0.<span class="fl">4.3</span>        utf8_1.<span class="fl">2.4</span>           </span>
<span id="cb132-29"><a href="multi-omics-integration.html#cb132-29" tabindex="-1"></a> [<span class="dv">22</span>] rmarkdown_2<span class="fl">.27</span>        ragg_1.<span class="fl">3.2</span>            purrr_1.<span class="fl">0.2</span>          </span>
<span id="cb132-30"><a href="multi-omics-integration.html#cb132-30" tabindex="-1"></a> [<span class="dv">25</span>] xfun_0<span class="fl">.46</span>             zlibbioc_1.<span class="fl">50.0</span>       beachmat_2.<span class="fl">20.0</span>      </span>
<span id="cb132-31"><a href="multi-omics-integration.html#cb132-31" tabindex="-1"></a> [<span class="dv">28</span>] jsonlite_1.<span class="fl">8.8</span>        DelayedArray_0.<span class="fl">30.1</span>   BiocParallel_1.<span class="fl">38.0</span>  </span>
<span id="cb132-32"><a href="multi-omics-integration.html#cb132-32" tabindex="-1"></a> [<span class="dv">31</span>] terra_1<span class="fl">.7</span><span class="dv">-78</span>          irlba_2.<span class="dv">3</span>.<span class="fl">5.1</span>         parallel_4.<span class="fl">4.1</span>       </span>
<span id="cb132-33"><a href="multi-omics-integration.html#cb132-33" tabindex="-1"></a> [<span class="dv">34</span>] R6_2.<span class="fl">5.1</span>              stringi_1.<span class="fl">8.4</span>         RColorBrewer_1<span class="fl">.1</span><span class="dv">-3</span>   </span>
<span id="cb132-34"><a href="multi-omics-integration.html#cb132-34" tabindex="-1"></a> [<span class="dv">37</span>] reticulate_1.<span class="fl">38.0</span>     parallelly_1.<span class="fl">37.1</span>     scattermore_1<span class="fl">.2</span>      </span>
<span id="cb132-35"><a href="multi-omics-integration.html#cb132-35" tabindex="-1"></a> [<span class="dv">40</span>] Rcpp_1.<span class="fl">0.13</span>           bookdown_0<span class="fl">.40</span>         knitr_1<span class="fl">.48</span>           </span>
<span id="cb132-36"><a href="multi-omics-integration.html#cb132-36" tabindex="-1"></a> [<span class="dv">43</span>] future.apply_1.<span class="fl">11.2</span>   R.utils_2.<span class="fl">12.3</span>        IRanges_2.<span class="fl">38.1</span>       </span>
<span id="cb132-37"><a href="multi-omics-integration.html#cb132-37" tabindex="-1"></a> [<span class="dv">46</span>] Matrix_1<span class="fl">.7</span><span class="dv">-0</span>          igraph_2.<span class="fl">0.3</span>          tidyselect_1.<span class="fl">2.1</span>     </span>
<span id="cb132-38"><a href="multi-omics-integration.html#cb132-38" tabindex="-1"></a> [<span class="dv">49</span>] rstudioapi_0.<span class="fl">16.0</span>     abind_1<span class="fl">.4</span><span class="dv">-5</span>           yaml_2.<span class="fl">3.9</span>           </span>
<span id="cb132-39"><a href="multi-omics-integration.html#cb132-39" tabindex="-1"></a> [<span class="dv">52</span>] codetools_0<span class="fl">.2</span><span class="dv">-20</span>      listenv_0.<span class="fl">9.1</span>         lattice_0<span class="fl">.22</span><span class="dv">-6</span>       </span>
<span id="cb132-40"><a href="multi-omics-integration.html#cb132-40" tabindex="-1"></a> [<span class="dv">55</span>] tibble_3.<span class="fl">2.1</span>          plyr_1.<span class="fl">8.9</span>            withr_3.<span class="fl">0.0</span>          </span>
<span id="cb132-41"><a href="multi-omics-integration.html#cb132-41" tabindex="-1"></a> [<span class="dv">58</span>] evaluate_0.<span class="fl">24.0</span>       future_1.<span class="fl">33.2</span>         pillar_1.<span class="fl">9.0</span>         </span>
<span id="cb132-42"><a href="multi-omics-integration.html#cb132-42" tabindex="-1"></a> [<span class="dv">61</span>] MatrixGenerics_1.<span class="fl">16.0</span> checkmate_2.<span class="fl">3.1</span>       stats4_4.<span class="fl">4.1</span>         </span>
<span id="cb132-43"><a href="multi-omics-integration.html#cb132-43" tabindex="-1"></a> [<span class="dv">64</span>] plotly_4.<span class="fl">10.4</span>         generics_0.<span class="fl">1.3</span>        dbscan_1<span class="fl">.2</span><span class="dv">-0</span>         </span>
<span id="cb132-44"><a href="multi-omics-integration.html#cb132-44" tabindex="-1"></a> [<span class="dv">67</span>] sp_2<span class="fl">.1</span><span class="dv">-4</span>              S4Vectors_0.<span class="fl">42.1</span>      ggplot2_3.<span class="fl">5.1</span>        </span>
<span id="cb132-45"><a href="multi-omics-integration.html#cb132-45" tabindex="-1"></a> [<span class="dv">70</span>] munsell_0.<span class="fl">5.1</span>         scales_1.<span class="fl">3.0</span>          gtools_3.<span class="fl">9.5</span>         </span>
<span id="cb132-46"><a href="multi-omics-integration.html#cb132-46" tabindex="-1"></a> [<span class="dv">73</span>] globals_0.<span class="fl">16.3</span>        glue_1.<span class="fl">7.0</span>            lazyeval_0.<span class="fl">2.2</span>       </span>
<span id="cb132-47"><a href="multi-omics-integration.html#cb132-47" tabindex="-1"></a> [<span class="dv">76</span>] tools_4.<span class="fl">4.1</span>           GiottoVisuals_0.<span class="fl">2.3</span>   data.table_1.<span class="fl">15.4</span>    </span>
<span id="cb132-48"><a href="multi-omics-integration.html#cb132-48" tabindex="-1"></a> [<span class="dv">79</span>] ScaledMatrix_1.<span class="fl">12.0</span>   cowplot_1.<span class="fl">1.3</span>         grid_4.<span class="fl">4.1</span>           </span>
<span id="cb132-49"><a href="multi-omics-integration.html#cb132-49" tabindex="-1"></a> [<span class="dv">82</span>] tidyr_1.<span class="fl">3.1</span>           colorspace_2<span class="fl">.1</span><span class="dv">-0</span>      BiocSingular_1.<span class="fl">20.0</span>  </span>
<span id="cb132-50"><a href="multi-omics-integration.html#cb132-50" tabindex="-1"></a> [<span class="dv">85</span>] cli_3.<span class="fl">6.3</span>             rsvd_1.<span class="fl">0.5</span>            textshaping_0.<span class="fl">4.0</span>    </span>
<span id="cb132-51"><a href="multi-omics-integration.html#cb132-51" tabindex="-1"></a> [<span class="dv">88</span>] fansi_1.<span class="fl">0.6</span>           S4Arrays_1.<span class="fl">4.1</span>        viridisLite_0.<span class="fl">4.2</span>    </span>
<span id="cb132-52"><a href="multi-omics-integration.html#cb132-52" tabindex="-1"></a> [<span class="dv">91</span>] dplyr_1.<span class="fl">1.4</span>           uwot_0.<span class="fl">2.2</span>            gtable_0.<span class="fl">3.5</span>         </span>
<span id="cb132-53"><a href="multi-omics-integration.html#cb132-53" tabindex="-1"></a> [<span class="dv">94</span>] R.methodsS3_1.<span class="fl">8.2</span>     digest_0.<span class="fl">6.36</span>         BiocGenerics_0.<span class="fl">50.0</span>  </span>
<span id="cb132-54"><a href="multi-omics-integration.html#cb132-54" tabindex="-1"></a> [<span class="dv">97</span>] SparseArray_1.<span class="fl">4.8</span>     ggrepel_0.<span class="fl">9.5</span>         htmlwidgets_1.<span class="fl">6.4</span>    </span>
<span id="cb132-55"><a href="multi-omics-integration.html#cb132-55" tabindex="-1"></a>[<span class="dv">100</span>] farver_2.<span class="fl">1.2</span>          htmltools_0.<span class="dv">5</span>.<span class="fl">8.1</span>     R.oo_1.<span class="fl">26.0</span>          </span>
<span id="cb132-56"><a href="multi-omics-integration.html#cb132-56" tabindex="-1"></a>[<span class="dv">103</span>] lifecycle_1.<span class="fl">0.4</span>       httr_1.<span class="fl">4.7</span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="image-transforms.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="interoperability-with-other-frameworks.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
