<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Xenium | Workshop: Spatial multi-omics data analysis with Giotto Suite</title>
  <meta name="description" content="9 Xenium | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Xenium | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="drieslab/giotto_workshop_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Xenium | Workshop: Spatial multi-omics data analysis with Giotto Suite" />
  
  
  

<meta name="author" content="Ruben Dries, Jiaji George Chen, Joselyn Cristina Chávez-Fuentes, Junxiang Xu , Edward Ruiz, Jeff Sheridan, Wen Wang" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visium-hd.html"/>
<link rel="next" href="spatial-proteomics-multiplex-if.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Giotto Workshop 2024</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#instructors"><i class="fa fa-check"></i><b>1.1</b> Instructors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#topics-and-schedule"><i class="fa fa-check"></i><b>1.2</b> Topics and Schedule:</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>1.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html"><i class="fa fa-check"></i><b>2</b> Spatial omics technologies</a>
<ul>
<li class="chapter" data-level="2.1" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#slides"><i class="fa fa-check"></i><b>2.1</b> Slides</a></li>
<li class="chapter" data-level="2.2" data-path="spatial-omics-technologies.html"><a href="spatial-omics-technologies.html#spatial-sequencing"><i class="fa fa-check"></i><b>2.2</b> Spatial sequencing</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html"><i class="fa fa-check"></i><b>3</b> Introduction to the Giotto package</a>
<ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#slides-1"><i class="fa fa-check"></i><b>3.1</b> Slides</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#ecosystem"><i class="fa fa-check"></i><b>3.2</b> Ecosystem</a></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#installation-python-environment"><i class="fa fa-check"></i><b>3.3</b> Installation + python environment</a></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-the-giotto-package.html"><a href="introduction-to-the-giotto-package.html#giotto-instructions"><i class="fa fa-check"></i><b>3.4</b> Giotto instructions</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html"><i class="fa fa-check"></i><b>4</b> Data formatting and Pre-processing</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#data-formats"><i class="fa fa-check"></i><b>4.1</b> Data formats</a></li>
<li class="chapter" data-level="4.2" data-path="data-formatting-and-pre-processing.html"><a href="data-formatting-and-pre-processing.html#pre-processing"><i class="fa fa-check"></i><b>4.2</b> Pre-processing</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html"><i class="fa fa-check"></i><b>5</b> Creating a Giotto object</a>
<ul>
<li class="chapter" data-level="5.1" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-matrix-locations"><i class="fa fa-check"></i><b>5.1</b> From matrix + locations</a></li>
<li class="chapter" data-level="5.2" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-subcellular-raw-data-transcripts-or-images-polygons"><i class="fa fa-check"></i><b>5.2</b> From subcellular raw data (transcripts or images) + polygons</a></li>
<li class="chapter" data-level="5.3" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#from-piece-wise"><i class="fa fa-check"></i><b>5.3</b> From piece-wise</a></li>
<li class="chapter" data-level="5.4" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#using-convenience-functions-for-popular-technologies-vizgen-xenium-cosmx"><i class="fa fa-check"></i><b>5.4</b> Using convenience functions for popular technologies (Vizgen, Xenium, CosMx, …)</a></li>
<li class="chapter" data-level="5.5" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#spatial-plots"><i class="fa fa-check"></i><b>5.5</b> Spatial plots</a></li>
<li class="chapter" data-level="5.6" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#subsetting"><i class="fa fa-check"></i><b>5.6</b> Subsetting</a></li>
<li class="chapter" data-level="5.7" data-path="creating-a-giotto-object.html"><a href="creating-a-giotto-object.html#mini-objects-giottodata"><i class="fa fa-check"></i><b>5.7</b> Mini objects &amp; GiottoData</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visium-part-i.html"><a href="visium-part-i.html"><i class="fa fa-check"></i><b>6</b> Visium Part I</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visium-part-i.html"><a href="visium-part-i.html#introduction-to-the-spatial-dataset"><i class="fa fa-check"></i><b>6.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="6.2" data-path="visium-part-i.html"><a href="visium-part-i.html#download-dataset"><i class="fa fa-check"></i><b>6.2</b> Download dataset</a></li>
<li class="chapter" data-level="6.3" data-path="visium-part-i.html"><a href="visium-part-i.html#create-the-giotto-object"><i class="fa fa-check"></i><b>6.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="6.4" data-path="visium-part-i.html"><a href="visium-part-i.html#subset-on-spots-that-were-covered-by-tissue"><i class="fa fa-check"></i><b>6.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="6.5" data-path="visium-part-i.html"><a href="visium-part-i.html#quality-control"><i class="fa fa-check"></i><b>6.5</b> Quality control</a></li>
<li class="chapter" data-level="6.6" data-path="visium-part-i.html"><a href="visium-part-i.html#filtering"><i class="fa fa-check"></i><b>6.6</b> Filtering</a></li>
<li class="chapter" data-level="6.7" data-path="visium-part-i.html"><a href="visium-part-i.html#normalization"><i class="fa fa-check"></i><b>6.7</b> Normalization</a></li>
<li class="chapter" data-level="6.8" data-path="visium-part-i.html"><a href="visium-part-i.html#feature-selection"><i class="fa fa-check"></i><b>6.8</b> Feature selection</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="visium-part-i.html"><a href="visium-part-i.html#highly-variable-features"><i class="fa fa-check"></i><b>6.8.1</b> Highly Variable Features:</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="visium-part-i.html"><a href="visium-part-i.html#dimension-reduction"><i class="fa fa-check"></i><b>6.9</b> Dimension Reduction</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="visium-part-i.html"><a href="visium-part-i.html#pca"><i class="fa fa-check"></i><b>6.9.1</b> PCA</a></li>
<li class="chapter" data-level="6.9.2" data-path="visium-part-i.html"><a href="visium-part-i.html#umap"><i class="fa fa-check"></i><b>6.9.2</b> UMAP</a></li>
<li class="chapter" data-level="6.9.3" data-path="visium-part-i.html"><a href="visium-part-i.html#t-sne"><i class="fa fa-check"></i><b>6.9.3</b> t-SNE</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="visium-part-i.html"><a href="visium-part-i.html#clustering"><i class="fa fa-check"></i><b>6.10</b> Clustering</a>
<ul>
<li class="chapter" data-level="6.10.1" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-leiden-clustering"><i class="fa fa-check"></i><b>6.10.1</b> Calculate Leiden clustering</a></li>
<li class="chapter" data-level="6.10.2" data-path="visium-part-i.html"><a href="visium-part-i.html#calculate-louvain-clustering"><i class="fa fa-check"></i><b>6.10.2</b> Calculate Louvain clustering</a></li>
</ul></li>
<li class="chapter" data-level="6.11" data-path="visium-part-i.html"><a href="visium-part-i.html#save-the-object"><i class="fa fa-check"></i><b>6.11</b> Save the object</a></li>
<li class="chapter" data-level="6.12" data-path="visium-part-i.html"><a href="visium-part-i.html#session-info"><i class="fa fa-check"></i><b>6.12</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="visium-part-ii.html"><a href="visium-part-ii.html"><i class="fa fa-check"></i><b>7</b> Visium Part II</a>
<ul>
<li class="chapter" data-level="7.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#load-the-object"><i class="fa fa-check"></i><b>7.1</b> Load the object</a></li>
<li class="chapter" data-level="7.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#differential-expression"><i class="fa fa-check"></i><b>7.2</b> Differential expression</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#gini-markers"><i class="fa fa-check"></i><b>7.2.1</b> Gini markers</a></li>
<li class="chapter" data-level="7.2.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#scran-markers"><i class="fa fa-check"></i><b>7.2.2</b> Scran markers</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="visium-part-ii.html"><a href="visium-part-ii.html#enrichment-deconvolution"><i class="fa fa-check"></i><b>7.3</b> Enrichment &amp; Deconvolution</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#pagerank"><i class="fa fa-check"></i><b>7.3.1</b> PAGE/Rank</a></li>
<li class="chapter" data-level="7.3.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatialdwls"><i class="fa fa-check"></i><b>7.3.2</b> SpatialDWLS</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-expression-patterns"><i class="fa fa-check"></i><b>7.4</b> Spatial expression patterns</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-variable-genes"><i class="fa fa-check"></i><b>7.4.1</b> Spatial variable genes</a></li>
<li class="chapter" data-level="7.4.2" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-co-expression-modules"><i class="fa fa-check"></i><b>7.4.2</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatially-informed-clusters"><i class="fa fa-check"></i><b>7.5</b> Spatially informed clusters</a></li>
<li class="chapter" data-level="7.6" data-path="visium-part-ii.html"><a href="visium-part-ii.html#spatial-domains-hmrf"><i class="fa fa-check"></i><b>7.6</b> Spatial domains HMRF</a></li>
<li class="chapter" data-level="7.7" data-path="visium-part-ii.html"><a href="visium-part-ii.html#interactive-tools"><i class="fa fa-check"></i><b>7.7</b> Interactive tools</a></li>
<li class="chapter" data-level="7.8" data-path="visium-part-ii.html"><a href="visium-part-ii.html#session-info-1"><i class="fa fa-check"></i><b>7.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="visium-hd.html"><a href="visium-hd.html"><i class="fa fa-check"></i><b>8</b> Visium HD</a>
<ul>
<li class="chapter" data-level="8.1" data-path="visium-hd.html"><a href="visium-hd.html#data-ingestion"><i class="fa fa-check"></i><b>8.1</b> Data Ingestion</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="visium-hd.html"><a href="visium-hd.html#read-in-raw-data"><i class="fa fa-check"></i><b>8.1.1</b> Read in raw data</a></li>
<li class="chapter" data-level="8.1.2" data-path="visium-hd.html"><a href="visium-hd.html#use-convenience-function"><i class="fa fa-check"></i><b>8.1.2</b> Use convenience function</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="visium-hd.html"><a href="visium-hd.html#tiling-and-aggregation"><i class="fa fa-check"></i><b>8.2</b> Tiling and aggregation</a></li>
<li class="chapter" data-level="8.3" data-path="visium-hd.html"><a href="visium-hd.html#scalability-and-projection-functions"><i class="fa fa-check"></i><b>8.3</b> Scalability and projection functions</a></li>
<li class="chapter" data-level="8.4" data-path="visium-hd.html"><a href="visium-hd.html#spatial-expression-patterns-1"><i class="fa fa-check"></i><b>8.4</b> Spatial expression patterns</a></li>
<li class="chapter" data-level="8.5" data-path="visium-hd.html"><a href="visium-hd.html#spatial-co-expression-modules-1"><i class="fa fa-check"></i><b>8.5</b> Spatial co-expression modules</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="xenium.html"><a href="xenium.html"><i class="fa fa-check"></i><b>9</b> Xenium</a>
<ul>
<li class="chapter" data-level="9.1" data-path="xenium.html"><a href="xenium.html#introduction-to-spatial-dataset"><i class="fa fa-check"></i><b>9.1</b> Introduction to spatial dataset</a></li>
<li class="chapter" data-level="9.2" data-path="xenium.html"><a href="xenium.html#additional-install-requirements"><i class="fa fa-check"></i><b>9.2</b> Additional install requirements</a></li>
<li class="chapter" data-level="9.3" data-path="xenium.html"><a href="xenium.html#read-in-raw-data-1"><i class="fa fa-check"></i><b>9.3</b> Read in raw data</a></li>
<li class="chapter" data-level="9.4" data-path="xenium.html"><a href="xenium.html#overlap-txs-polygons"><i class="fa fa-check"></i><b>9.4</b> Overlap txs &amp; polygons</a></li>
<li class="chapter" data-level="9.5" data-path="xenium.html"><a href="xenium.html#transcript-enrichment-gsea"><i class="fa fa-check"></i><b>9.5</b> Transcript enrichment GSEA</a></li>
<li class="chapter" data-level="9.6" data-path="xenium.html"><a href="xenium.html#spatial-location-analysis"><i class="fa fa-check"></i><b>9.6</b> Spatial location analysis</a></li>
<li class="chapter" data-level="9.7" data-path="xenium.html"><a href="xenium.html#visualizations"><i class="fa fa-check"></i><b>9.7</b> Visualizations</a></li>
<li class="chapter" data-level="9.8" data-path="xenium.html"><a href="xenium.html#exploration"><i class="fa fa-check"></i><b>9.8</b> —– exploration</a></li>
<li class="chapter" data-level="9.9" data-path="xenium.html"><a href="xenium.html#image"><i class="fa fa-check"></i><b>9.9</b> IMAGE</a></li>
<li class="chapter" data-level="9.10" data-path="xenium.html"><a href="xenium.html#ometif_2_tifimage_path-output_file-out_tif-chunk_rows-chunk_rows-overwrite-true"><i class="fa fa-check"></i><b>9.10</b> ometif_2_tif(image_path, output_file = out_tif, chunk_rows = chunk_rows, overwrite = TRUE)</a></li>
<li class="chapter" data-level="9.11" data-path="xenium.html"><a href="xenium.html#example-usage"><i class="fa fa-check"></i><b>9.11</b> Example usage</a></li>
<li class="chapter" data-level="9.12" data-path="xenium.html"><a href="xenium.html#transcripts"><i class="fa fa-check"></i><b>9.12</b> TRANSCRIPTS</a></li>
<li class="chapter" data-level="9.13" data-path="xenium.html"><a href="xenium.html#where-are-the-issue-tx-at"><i class="fa fa-check"></i><b>9.13</b> where are the issue TX at?</a></li>
<li class="chapter" data-level="9.14" data-path="xenium.html"><a href="xenium.html#polys"><i class="fa fa-check"></i><b>9.14</b> POLYS</a></li>
<li class="chapter" data-level="9.15" data-path="xenium.html"><a href="xenium.html#giotto"><i class="fa fa-check"></i><b>9.15</b> giotto</a></li>
<li class="chapter" data-level="9.16" data-path="xenium.html"><a href="xenium.html#no-hvf-calc-since-there-are-not-that-many-different-gene-species"><i class="fa fa-check"></i><b>9.16</b> no HVF calc since there are not that many different gene species</a></li>
<li class="chapter" data-level="9.17" data-path="xenium.html"><a href="xenium.html#set-seed-for-reproducibility"><i class="fa fa-check"></i><b>9.17</b> set seed for reproducibility</a></li>
<li class="chapter" data-level="9.18" data-path="xenium.html"><a href="xenium.html#seems-like-a-good-res"><i class="fa fa-check"></i><b>9.18</b> 1.4 seems like a good res</a></li>
<li class="chapter" data-level="9.19" data-path="xenium.html"><a href="xenium.html#niches"><i class="fa fa-check"></i><b>9.19</b> niches</a></li>
<li class="chapter" data-level="9.20" data-path="xenium.html"><a href="xenium.html#visualize-niche-level-enrichment-for-leiden-cluster-3-in-spat_unit-cell"><i class="fa fa-check"></i><b>9.20</b> visualize niche-level enrichment for leiden cluster 3 in spat_unit “cell”</a></li>
<li class="chapter" data-level="9.21" data-path="xenium.html"><a href="xenium.html#add-kmeans-clustering-of-niches-to-cell-metadata"><i class="fa fa-check"></i><b>9.21</b> add kmeans clustering of niches to cell metadata</a></li>
<li class="chapter" data-level="9.22" data-path="xenium.html"><a href="xenium.html#spatially-visualize-the-niches"><i class="fa fa-check"></i><b>9.22</b> Spatially visualize the niches</a></li>
<li class="chapter" data-level="9.23" data-path="xenium.html"><a href="xenium.html#gene-enrichment"><i class="fa fa-check"></i><b>9.23</b> Gene enrichment</a></li>
<li class="chapter" data-level="9.24" data-path="xenium.html"><a href="xenium.html#violinplot"><i class="fa fa-check"></i><b>9.24</b> violinplot</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html"><i class="fa fa-check"></i><b>10</b> Spatial proteomics: multiplex IF</a>
<ul>
<li class="chapter" data-level="10.1" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#read-in-raw-data-2"><i class="fa fa-check"></i><b>10.1</b> Read in raw data</a></li>
<li class="chapter" data-level="10.2" data-path="spatial-proteomics-multiplex-if.html"><a href="spatial-proteomics-multiplex-if.html#overlap-intensity-workflows"><i class="fa fa-check"></i><b>10.2</b> Overlap intensity &amp; workflows</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html"><i class="fa fa-check"></i><b>11</b> Working with multiple samples</a>
<ul>
<li class="chapter" data-level="11.1" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#create-individual-giotto-objects"><i class="fa fa-check"></i><b>11.1</b> Create individual giotto objects</a></li>
<li class="chapter" data-level="11.2" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#join-giotto-objects"><i class="fa fa-check"></i><b>11.2</b> Join Giotto Objects</a></li>
<li class="chapter" data-level="11.3" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#perform-harmony-and-default-workflows"><i class="fa fa-check"></i><b>11.3</b> Perform Harmony and default workflows</a></li>
<li class="chapter" data-level="11.4" data-path="working-with-multiple-samples.html"><a href="working-with-multiple-samples.html#visualizations-1"><i class="fa fa-check"></i><b>11.4</b> Visualizations</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html"><i class="fa fa-check"></i><b>12</b> Spatial multi-modal analysis</a>
<ul>
<li class="chapter" data-level="12.1" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#co-registration-of-datasets"><i class="fa fa-check"></i><b>12.1</b> Co-registration of datasets</a></li>
<li class="chapter" data-level="12.2" data-path="spatial-multi-modal-analysis.html"><a href="spatial-multi-modal-analysis.html#examples-in-giotto-suite-manuscript"><i class="fa fa-check"></i><b>12.2</b> Examples in giotto suite manuscript</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html"><i class="fa fa-check"></i><b>13</b> Multi-omics integration</a>
<ul>
<li class="chapter" data-level="13.1" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#introduction-to-the-spatial-dataset-1"><i class="fa fa-check"></i><b>13.1</b> Introduction to the spatial dataset</a></li>
<li class="chapter" data-level="13.2" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#download-dataset-1"><i class="fa fa-check"></i><b>13.2</b> Download dataset</a></li>
<li class="chapter" data-level="13.3" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#create-the-giotto-object-1"><i class="fa fa-check"></i><b>13.3</b> Create the Giotto object</a></li>
<li class="chapter" data-level="13.4" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#subset-on-spots-that-were-covered-by-tissue-1"><i class="fa fa-check"></i><b>13.4</b> Subset on spots that were covered by tissue</a></li>
<li class="chapter" data-level="13.5" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#rna-processing"><i class="fa fa-check"></i><b>13.5</b> RNA processing</a></li>
<li class="chapter" data-level="13.6" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#protein-processing"><i class="fa fa-check"></i><b>13.6</b> Protein processing</a></li>
<li class="chapter" data-level="13.7" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#multi-omics-integration-1"><i class="fa fa-check"></i><b>13.7</b> Multi-omics integration</a></li>
<li class="chapter" data-level="13.8" data-path="multi-omics-integration.html"><a href="multi-omics-integration.html#session-info-2"><i class="fa fa-check"></i><b>13.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html"><i class="fa fa-check"></i><b>14</b> Interoperability with other frameworks</a>
<ul>
<li class="chapter" data-level="14.1" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#anndataspatialdata"><i class="fa fa-check"></i><b>14.1</b> AnnData/SpatialData</a></li>
<li class="chapter" data-level="14.2" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#spatialexperiment"><i class="fa fa-check"></i><b>14.2</b> SpatialExperiment</a></li>
<li class="chapter" data-level="14.3" data-path="interoperability-with-other-frameworks.html"><a href="interoperability-with-other-frameworks.html#seurat"><i class="fa fa-check"></i><b>14.3</b> Seurat</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html"><i class="fa fa-check"></i><b>15</b> Interoperability with isolated tools</a>
<ul>
<li class="chapter" data-level="15.1" data-path="interoperability-with-isolated-tools.html"><a href="interoperability-with-isolated-tools.html#spatial-niche-trajectory-analysis"><i class="fa fa-check"></i><b>15.1</b> Spatial niche trajectory analysis</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html"><i class="fa fa-check"></i><b>16</b> Interactivity with the R/Spatial ecosystem</a>
<ul>
<li class="chapter" data-level="16.1" data-path="interactivity-with-the-rspatial-ecosystem.html"><a href="interactivity-with-the-rspatial-ecosystem.html#kriging"><i class="fa fa-check"></i><b>16.1</b> Kriging</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html"><i class="fa fa-check"></i><b>17</b> Contributing to Giotto</a>
<ul>
<li class="chapter" data-level="17.1" data-path="contributing-to-giotto.html"><a href="contributing-to-giotto.html#contribution-guideline"><i class="fa fa-check"></i><b>17.1</b> Contribution guideline</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workshop: Spatial multi-omics data analysis with Giotto Suite</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="xenium" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">9</span> Xenium<a href="xenium.html#xenium" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Jiaji George Chen</p>
<p>August 6th 2024</p>
<div id="introduction-to-spatial-dataset" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Introduction to spatial dataset<a href="xenium.html#introduction-to-spatial-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="img/02_session3/Xenium-General-Workflow-1024x709.png" width="80%" style="display: block; margin: auto;" /></p>
<p>This is the 10X Xenium FFPE Human Lung Cancer dataset. Xenium captures
individual transcript detections with a spatial resolution of 100s of
nanometers, providing an extremely highly resolved subcellular spatial
dataset. This particular dataset also showcases their newly released
multimodal cell segmentation outputs.</p>
<p>Previous Xenium datasets provided nuclear segmentations based on DAPI
then generated cell</p>
<p>The Xenium Human Multi-Tissue and Cancer Panel (377) genes was used. The
exported data is from their Xenium Onboard Analysis v2.0.0 pipeline.</p>
<p>The data to run this tutorial can be found
<a href="https://www.10xgenomics.com/datasets/preview-data-ffpe-human-lung-cancer-with-xenium-multimodal-cell-segmentation-1-standard">here</a></p>
<p>The files can be downloaded by running the following in the bash
terminal:</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb100-1"><a href="xenium.html#cb100-1" tabindex="-1"></a><span class="co"># Input Files</span></span>
<span id="cb100-2"><a href="xenium.html#cb100-2" tabindex="-1"></a></span>
<span id="cb100-3"><a href="xenium.html#cb100-3" tabindex="-1"></a><span class="fu">wget</span> https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_humanLung_Cancer_FFPE/Xenium_V1_humanLung_Cancer_FFPE_gene_panel.json</span>
<span id="cb100-4"><a href="xenium.html#cb100-4" tabindex="-1"></a><span class="fu">wget</span> https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_humanLung_Cancer_FFPE/Xenium_V1_humanLung_Cancer_FFPE_he_image.ome.tif</span>
<span id="cb100-5"><a href="xenium.html#cb100-5" tabindex="-1"></a><span class="fu">wget</span> https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_humanLung_Cancer_FFPE/Xenium_V1_humanLung_Cancer_FFPE_he_imagealignment.csv</span>
<span id="cb100-6"><a href="xenium.html#cb100-6" tabindex="-1"></a></span>
<span id="cb100-7"><a href="xenium.html#cb100-7" tabindex="-1"></a><span class="co"># Output Files</span></span>
<span id="cb100-8"><a href="xenium.html#cb100-8" tabindex="-1"></a><span class="fu">wget</span> https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_humanLung_Cancer_FFPE/Xenium_V1_humanLung_Cancer_FFPE_outs.zip</span>
<span id="cb100-9"><a href="xenium.html#cb100-9" tabindex="-1"></a><span class="fu">wget</span> https://cf.10xgenomics.com/samples/xenium/2.0.0/Xenium_V1_humanLung_Cancer_FFPE/Xenium_V1_humanLung_Cancer_FFPE_xe_outs.zip</span></code></pre></div>
<ul>
<li><em>Input Files</em> are additional files that are not part of the output
bundles, usually some raw files the pipeline uses or extra info.</li>
<li><em>Output Files</em> are the directories that are exported by the Xenium
pipeline that can be expected from every run.</li>
</ul>
</div>
<div id="additional-install-requirements" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Additional install requirements<a href="xenium.html#additional-install-requirements" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>arrow is needed</li>
<li>requires ZTSD codec to open these parquets.</li>
</ul>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="xenium.html#cb101-1" tabindex="-1"></a>save_dir <span class="ot">&lt;-</span> <span class="st">&quot;~/Documents/GitHub/giotto_workshop_2024/img/02_session3&quot;</span></span></code></pre></div>
</div>
<div id="read-in-raw-data-1" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Read in raw data<a href="xenium.html#read-in-raw-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p>Transcript coordinates</p></li>
<li><p>Polygon coordinates</p></li>
<li><p>Visualizations</p></li>
<li><p>affine mapping?</p></li>
</ul>
</div>
<div id="overlap-txs-polygons" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Overlap txs &amp; polygons<a href="xenium.html#overlap-txs-polygons" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Typical aggregated workflow</li>
<li>Feature/molecule specific analysis</li>
<li>Visualizations</li>
</ul>
</div>
<div id="transcript-enrichment-gsea" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Transcript enrichment GSEA<a href="xenium.html#transcript-enrichment-gsea" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>text</p>
</div>
<div id="spatial-location-analysis" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> Spatial location analysis<a href="xenium.html#spatial-location-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Spatial cell type co-localization analysis</li>
<li>Spatial niche analysis</li>
<li>Spatial niche trajectory analysis</li>
</ul>
</div>
<div id="visualizations" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> Visualizations<a href="xenium.html#visualizations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>zarr_exp &lt;- “<a href="https://scc-ondemand1.bu.edu/pun/sys/dashboard/files/fs//projectnb/rd-spat/DATA/Public_data/Spatial/Multiplexing_RNA/xenium/v2.0_pipeline/FFPE_LungCancer/cell_features" class="uri">https://scc-ondemand1.bu.edu/pun/sys/dashboard/files/fs//projectnb/rd-spat/DATA/Public_data/Spatial/Multiplexing_RNA/xenium/v2.0_pipeline/FFPE_LungCancer/cell_features</a>”</p>
<p>datadir &lt;- “/projectnb2/rd-spat/DATA/Public_data/Spatial/Multiplexing_RNA/xenium/v2.0_pipeline/FFPE_LungCancer/”</p>
<p>manifest &lt;- list.files(
datadir, full.names = TRUE,
) |&gt; as.list()
names(manifest) &lt;- list.files(datadir)
force(names(manifest))</p>
<p>tx_path &lt;- manifest<span class="math inline">\(transcripts.csv.gz
poly_cell_path &lt;- manifest\)</span>cell_boundaries.csv.gz
poly_nuc_path &lt;- manifest$nucleus_boundaries.csv.gz</p>
<p>image_path &lt;- manifest<span class="math inline">\(Xenium_V1_humanLung_Cancer_FFPE_he_image.ome.tif
aff_path &lt;- manifest\)</span>Xenium_V1_humanLung_Cancer_FFPE_he_imagealignment.csv</p>
<p>cell &lt;- manifest$cells.csv.gz</p>
</div>
<div id="exploration" class="section level2 hasAnchor" number="9.8">
<h2><span class="header-section-number">9.8</span> —– exploration<a href="xenium.html#exploration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="image" class="section level2 hasAnchor" number="9.9">
<h2><span class="header-section-number">9.9</span> IMAGE<a href="xenium.html#image" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>aff &lt;- data.table::fread(aff_path)</p>
<p>reticulate::source_python(file = “/projectnb/rd-spat/HOME/george/scripts/tifffile_convert.py”)
GiottoUtils::package_check(“os”, repository = “pip:os”)
GiottoUtils::package_check(“multiprocessing”, repository = “pip:multiprocessing”)
GiottoUtils::package_check(“tifffile”, repository = “pip:tifffile”)</p>
<p>out_tif &lt;- “/projectnb/rd-spat/HOME/george/tmp/tifffile_testing/test.tif”</p>
</div>
<div id="ometif_2_tifimage_path-output_file-out_tif-chunk_rows-chunk_rows-overwrite-true" class="section level2 hasAnchor" number="9.10">
<h2><span class="header-section-number">9.10</span> ometif_2_tif(image_path, output_file = out_tif, chunk_rows = chunk_rows, overwrite = TRUE)<a href="xenium.html#ometif_2_tifimage_path-output_file-out_tif-chunk_rows-chunk_rows-overwrite-true" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>ometif_2_tif(image_path, output_file = out_tif, overwrite = TRUE)
test &lt;- createGiottoLargeImage(out_tif)</p>
<p>n_before &lt;- .nr(tx_arrow)
tx_arrow &lt;- dplyr::filter(tx_arrow, qv &gt; qv_threshold)
n_after &lt;- .nr(tx_arrow)</p>
<p>out_if_fmt &lt;- “/projectnb/rd-spat/HOME/george/tmp/tifffile_testing/if%d.tif”
source_if_fmt &lt;- “/projectnb2/rd-spat/DATA/Public_data/Spatial/Multiplexing_RNA/xenium/v2.0_pipeline/FFPE_LungCancer//morphology_focus/morphology_focus_000%d.ome.tif”</p>
<p>ometif_2_tif(sprintf(source_if_fmt, 0), output_file = sprintf(out_if_fmt, 0), overwrite = TRUE)
ometif_2_tif(sprintf(source_if_fmt, 1), output_file = sprintf(out_if_fmt, 1), overwrite = TRUE)
ometif_2_tif(sprintf(source_if_fmt, 2), output_file = sprintf(out_if_fmt, 2), overwrite = TRUE)
ometif_2_tif(sprintf(source_if_fmt, 3), output_file = sprintf(out_if_fmt, 3), overwrite = TRUE)</p>
<p>img_if_list &lt;- lapply(
sprintf(out_if_fmt, 0:3),
createGiottoLargeImage
)</p>
</div>
<div id="example-usage" class="section level2 hasAnchor" number="9.11">
<h2><span class="header-section-number">9.11</span> Example usage<a href="xenium.html#example-usage" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>input_file = “path/to/input/file.ome.tif”
output_file = “path/to/output/file.tif”
chunk_size = (1000, 1000) # Adjust the chunk size as needed
num_processes = 4 # Adjust the number of processes based on available CPU cores</p>
<p>ometif_2_tif(input_file, output_file, chunk_size, num_processes)</p>
</div>
<div id="transcripts" class="section level2 hasAnchor" number="9.12">
<h2><span class="header-section-number">9.12</span> TRANSCRIPTS<a href="xenium.html#transcripts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>tx &lt;- data.table::fread(tx_path, colClasses = c(transcript_id = “character”))
poly_cell_dt &lt;- data.table::fread(poly_cell_path)
poly_nuc_dt &lt;- data.table::fread(poly_nuc_path)</p>
<p>gpoints_list &lt;- GiottoClass::createGiottoPoints(
tx,
x_colname = “x_location”,
y_colname = “y_location”,
feat_ID_colname = “feature_name”,
feat_type = c(“rna”, “NegControlProbe”, “UnassignedCodeword”, “NegControlCodeword”),
split_keyword = list(“NegControlProbe”, “UnassignedCodeword”, “NegControlCodeword”)
)</p>
<p>gp_filter &lt;- gpoints_list<span class="math inline">\(rna[gpoints_list\)</span>rna$qv &gt;= 20]</p>
</div>
<div id="where-are-the-issue-tx-at" class="section level2 hasAnchor" number="9.13">
<h2><span class="header-section-number">9.13</span> where are the issue TX at?<a href="xenium.html#where-are-the-issue-tx-at" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>plot(<a href="mailto:gp_filter@spatVector" class="email">gp_filter@spatVector</a>, values = gp_filter$qv, type = “continuous”, cex = 0.1, col = GiottoVisuals::getColors(“magma”))</p>
<p>library(ggplot2)
type_qv &lt;- data.table::rbindlist(
list(
data.table::data.table(type = “genes”, qv = gpoints_list<span class="math inline">\(rna\)</span>qv),
data.table::data.table(type = “NegControlProbe”, qv = gpoints_list<span class="math inline">\(NegControlProbe\)</span>qv),
data.table::data.table(type = “UnassignedCodeword”, qv = gpoints_list<span class="math inline">\(UnassignedCodeword\)</span>qv),
data.table::data.table(type = “NegControlCodeword”, qv = gpoints_list<span class="math inline">\(NegControlCodeword\)</span>qv)
)
)
ggplot(data = type_qv, aes(x = type, y = qv, fill = type)) +
geom_violin()</p>
</div>
<div id="polys" class="section level2 hasAnchor" number="9.14">
<h2><span class="header-section-number">9.14</span> POLYS<a href="xenium.html#polys" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>gpoly_cell &lt;- createGiottoPolygon(poly_cell_dt, name = “cell”)
gpoly_nuc &lt;- createGiottoPolygon(poly_nuc_dt, name = “nucleus”)</p>
<p>mini_ext &lt;- ext(6500,7500, 1300, 2000)</p>
<p>plot(gpoly_cell, col = “magenta”, ext = mini_ext, background = “black”)
plot(gpoly_nuc, add = T, col = “cyan”, ext = mini_ext)</p>
<p>plot(gp_filter, add = T, col = “yellow”, alpha = 0.3, ext = mini_ext, raster = F)</p>
</div>
<div id="giotto" class="section level2 hasAnchor" number="9.15">
<h2><span class="header-section-number">9.15</span> giotto<a href="xenium.html#giotto" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>g &lt;- giotto()
g &lt;- setGiotto(g, gp_filter)
g &lt;- setGiotto(g, gpoly_cell)
g &lt;- setGiotto(g, gpoly_nuc)</p>
<p>g &lt;- calculateOverlap(
g,
feat_info = “rna”,
spatial_info = “cell”,
return_gobject = T
)</p>
<p>g &lt;- overlapToMatrix(
g,
poly_info = “cell”,
feat_info = “rna”,
return_gobject = TRUE
)</p>
<p>savedir &lt;- “/projectnb2/rd-spat/HOME/george/projects/xenium_processing/lung_test/”
instructions(g, “save_dir”) &lt;- savedir
instructions(g, “save_plot”) &lt;- TRUE
instructions(g, “return_plot”) &lt;- FALSE
instructions(g, “show_plot”) &lt;- FALSE</p>
<p>g &lt;- addStatistics(g, expression_values = “raw”)
spatPlot2D(g, cell_color = “total_expr”, color_as_factor = FALSE,
point_shape = “no_border”, point_size = 0.1, gradient_style = “s”,
background_color = “black”,
save_param = list(
base_width = 15,
base_height = 6
))</p>
<p>hist(pDataDT(g)$nr_feats)</p>
<p>filterCombinations(
g,
feat_det_in_min_cells = c(100, 200, 200),
min_det_feats_per_cell = c(10, 20, 30),
expression_thresholds = 1
)</p>
<p>g &lt;- filterGiotto(
g,
feat_det_in_min_cells = 100,
min_det_feats_per_cell = 10,
expression_threshold = 1
)</p>
<p>g &lt;- normalizeGiotto(g)</p>
</div>
<div id="no-hvf-calc-since-there-are-not-that-many-different-gene-species" class="section level2 hasAnchor" number="9.16">
<h2><span class="header-section-number">9.16</span> no HVF calc since there are not that many different gene species<a href="xenium.html#no-hvf-calc-since-there-are-not-that-many-different-gene-species" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>g &lt;- runPCA(gobject = g,
spat_unit = ‘cell’,
expression_values = ‘scaled’,
feats_to_use = NULL,
scale_unit = F,
center = F)</p>
<p>screePlot(g, ncp = 20)</p>
<p>my_colors &lt;- c(“#6D0C00”, “magenta”, “#FFBBCE”, “#FFDDDF”, “#FFEEEE”, “#FFFEFE”, “white”, “white”)</p>
<p>plotPCA(g,
cell_color = “total_expr”,
color_as_factor = FALSE,
cell_color_gradient = my_colors,
point_shape = “no_border”,
gradient_style = “s”,
background = “black”,
point_size = 0.1,
save_param = list(
save_name = “pca_expr”
)
)</p>
<p>g &lt;- runUMAPprojection(
g, dimensions_to_use = 1:15,
random_subset = 10000
)</p>
<p>dimPlot2D(
g,
dim_reduction_name = “umap.projection”,
dim_reduction_to_use = “umap”,
cell_color = “total_expr”,
cell_color_gradient = my_colors,
color_as_factor = FALSE,
point_shape = “no_border”,
gradient_style = “s”,
background = “black”,
point_size = 0.1,
save_param = list(
save_name = “umap.projection_expr”
)
)</p>
<p>g &lt;- createNearestNetwork(g)</p>
</div>
<div id="set-seed-for-reproducibility" class="section level2 hasAnchor" number="9.17">
<h2><span class="header-section-number">9.17</span> set seed for reproducibility<a href="xenium.html#set-seed-for-reproducibility" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>set.seed(123)
g_mini &lt;- subsetGiotto(g, cell_ids = sample(spatIDs(g), size = 10000))</p>
<p>g_mini &lt;- createNearestNetwork(g_mini)</p>
<p>g_mini &lt;- doLeidenCluster(g_mini, resolution = 1.4)</p>
<p>dimPlot2D(
g_mini,
spat_unit = “cell”,
feat_type = “rna”,
cell_color = “leiden_clus”,
dim_reduction_name = “umap.projection”,
dim_reduction_to_use = “umap”,
save_param = list(
save_name = “mini_umap_1.2”
)
)</p>
<p>clus_color &lt;- getColors(“Vivid”, n = 17)[]
clus_color[9] &lt;- “lightgrey”
clus_color[5] &lt;- “yellow”
clus_color[3] &lt;- “#00AAFF”
clus_color[6] &lt;- “darkgreen”
clus_color &lt;- clus_color[c(2, 5, 1, 7, 6, 3, 4, 9, 8, 10, 12, 11, 14, 13, 15, 16, 17)]</p>
<p>clus_color &lt;- c(“#00AAFF”, “#E58606”, “#B763A7”, “#437478”, “#87727B”,
“darkgreen”, “#56BC9D”, “#5668AF”, “yellow”, “#B99F3A”,
“#E5625E”, “#448DAE”, “#A5AA99”, “#D8497A”, “#93538D”,
“#BF5C91”, “lightgrey”)</p>
<p>dimPlot2D(
g_mini,
spat_unit = “cell”,
feat_type = “rna”,
cell_color = “leiden_clus”,
cell_color_code = clus_color,
dim_reduction_name = “umap.projection”,
dim_reduction_to_use = “umap”,
save_param = list(
save_name = “mini_umap_1.4”
)
)</p>
</div>
<div id="seems-like-a-good-res" class="section level2 hasAnchor" number="9.18">
<h2><span class="header-section-number">9.18</span> 1.4 seems like a good res<a href="xenium.html#seems-like-a-good-res" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>project values back via kNN classifier (FNN::knn)</p>
<p>g &lt;- doClusterProjection(
target_gobject = g,
target_cluster_label_name = “leiden_clus”,
source_gobject = g_mini, # res 1.4
source_cluster_labels = “leiden_clus”
)</p>
<p>dimPlot2D(g, cell_color = “leiden_clus”,
cell_color_code = clus_color[c(1, 10:17, 2:9)],
dim_reduction_to_use = “umap”,
dim_reduction_name = “umap.projection”,
save_param = list(
save_name = “umap.project_1.4”
))</p>
<p>cellmeta &lt;- pDataDT(g)
cellmeta[, major_clus := leiden_clus]
cellmeta[major_clus %in% c(1, 7, 6, 8, 3, 5), major_clus := “combined_1”]
m_clus &lt;- cellmeta[, c(“cell_ID”, “major_clus”)]
g &lt;- addCellMetadata(g, new_metadata = m_clus, by_column = T)</p>
<p>spatPlot2D(g, cell_color = “leiden_clus”,
point_size = 0.1,
point_shape = “no_border”,
cell_color_code = clus_color[c(1, 10:17, 2:9)],
background = “black”,
save_param = list(
base_width = 15,
base_height = 6,
save_name = “spat_leiden”
))</p>
<p>spatPlot2D(g, cell_color = “major_clus”,
point_size = 0.1,
point_shape = “no_border”,
cell_color_code = clus_color[c(1, 10:17, 2:9)][c(1:3,5:9, 14, 17, 11)],
background = “black”,
save_param = list(
base_width = 15,
base_height = 6,
save_name = “spat_major”
))</p>
<p>spatFeatPlot2D(g, feats = c(“MKI67”, “CD3E”, “CD4”, “CD8A”, “CD19”, “CD79A”),
point_size = 0.1,
point_shape = “no_border”,
background = “black”,
cow_n_col = 1L,
gradient_style = “s”,
save_param = list(
base_width = 15,
base_height = 30
))</p>
<p>spatFeatPlot2D(g, feats = c(“CD68”, “CD27”, “SFTA2”, “KRT7”, “SERPINB3”, “SOX2”, “SMOC”),
point_size = 0.1,
point_shape = “no_border”,
background = “black”,
cow_n_col = 1L,
gradient_style = “s”,
save_param = list(
base_width = 15,
base_height = 30
))</p>
</div>
<div id="niches" class="section level2 hasAnchor" number="9.19">
<h2><span class="header-section-number">9.19</span> niches<a href="xenium.html#niches" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>g = createSpatialNetwork(
gobject = g,
spat_unit = ‘cell’,
method = ‘Delaunay’
)</p>
<p>g = calculateSpatCellMetadataProportions(
gobject = g,
spat_unit = ‘cell’,
feat_type = ‘rna’,
spat_network = ‘Delaunay_network’,
metadata_column = ‘leiden_clus’,
name = ‘proportion’
)</p>
</div>
<div id="visualize-niche-level-enrichment-for-leiden-cluster-3-in-spat_unit-cell" class="section level2 hasAnchor" number="9.20">
<h2><span class="header-section-number">9.20</span> visualize niche-level enrichment for leiden cluster 3 in spat_unit “cell”<a href="xenium.html#visualize-niche-level-enrichment-for-leiden-cluster-3-in-spat_unit-cell" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>spatPlot2D(
gobject = g,
spat_unit = ‘cell’,
point_size = 0.1,
spat_enr_names = ‘proportion’,
color_as_factor = FALSE,
gradient_style = ‘sequential’,
cell_color = ‘3’,
point_shape = “no_border”,
save_param = list(
save_name = “niche_3”,
base_width = 15,
base_height = 6
)
)</p>
<p>prop_table = getSpatialEnrichment(
g, spat_unit = ‘cell’, name = ‘proportion’, output = ‘data.table’
)</p>
<p>convert the data.table to a sparse Matrix with row and colnames
here we use a utility function to perform the operation</p>
<p>prop_matrix = GiottoUtils::dt_to_matrix(prop_table)</p>
<p>These enrichments are essentially a measure of how many cells of each
leiden cluster exist in the local region</p>
<p>Using kmeans, we can classify each cell by its niche leiden cluster proportions</p>
<p>set.seed(12345) # set seed for kmeans
prop_kmeans = kmeans(x = prop_matrix, centers = 6, iter.max = 100, nstart = 3)
prop_kmeansDT = data.table::data.table(
cell_ID = names(prop_kmeans<span class="math inline">\(cluster),
  niche = prop_kmeans\)</span>cluster
)</p>
</div>
<div id="add-kmeans-clustering-of-niches-to-cell-metadata" class="section level2 hasAnchor" number="9.21">
<h2><span class="header-section-number">9.21</span> add kmeans clustering of niches to cell metadata<a href="xenium.html#add-kmeans-clustering-of-niches-to-cell-metadata" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>g = addCellMetadata(
g, spat_unit = ‘cell’ ,
new_metadata = prop_kmeansDT,
by_column = TRUE,
column_cell_ID = ‘cell_ID’
)</p>
</div>
<div id="spatially-visualize-the-niches" class="section level2 hasAnchor" number="9.22">
<h2><span class="header-section-number">9.22</span> Spatially visualize the niches<a href="xenium.html#spatially-visualize-the-niches" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>spatPlot(gobject = g,
show_network = TRUE,
network_color = ‘lightgray’,
spatial_network_name = ‘Delaunay_network’,
cell_color = ‘niche’,
point_size = 0.1,
point_shape = “no_border”,
save_param = list(
save_name = “niches”,
base_width = 15,
base_height = 6
))</p>
</div>
<div id="gene-enrichment" class="section level2 hasAnchor" number="9.23">
<h2><span class="header-section-number">9.23</span> Gene enrichment<a href="xenium.html#gene-enrichment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>scran_markers &lt;- findScranMarkers_one_vs_all(
g,
cluster_column = “leiden_clus”,
expression_values = “normalized”
)</p>
<p>topgenes_scran = scran_markers[, head(.SD, 2), by = ‘cluster’]$feats</p>
</div>
<div id="violinplot" class="section level2 hasAnchor" number="9.24">
<h2><span class="header-section-number">9.24</span> violinplot<a href="xenium.html#violinplot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>violinPlot(g,
feats = unique(topgenes_scran),
cluster_column = ‘leiden_clus’,
strip_text = 10, strip_position = ‘right’,
save_param = list(base_width = 5))</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visium-hd.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="spatial-proteomics-multiplex-if.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
